



<!doctype html>
<html lang="en" class="no-js">
  <head>
    <script src="//assets.adobedtm.com/launch-EN942f79c83ff84c10a45b4856a85eaccb.min.js" async></script>
    <script>!function(e,a,t,n,g,c,o){e.GoogleAnalyticsObject=g,e.ga=e.ga||function(){(e.ga.q=e.ga.q||[]).push(arguments)},e.ga.l=1*new Date,c=a.createElement(t),o=a.getElementsByTagName(t)[0],c.async=1,c.src="https://www.google-analytics.com/analytics.js",o.parentNode.insertBefore(c,o)}(window,document,"script",0,"ga"),ga("create","UA-107924368-1","auto"),ga("send","pageview")</script>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../../images/favicon.ico">
      <meta name="generator" content="mkdocs-1.1, mkdocs-material-4.6.3">
    
    
      
        <title>Organizing Data into a Study View - St. Jude Cloud Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../../assets/stylesheets/application.7d2ef1ff.css">
      
      
    
    
      <script src="../../../../assets/javascripts/modernizr.eca31fed.js"></script>
    
    
      
    
    <link rel="stylesheet" href="../../../../assets/fonts/material-icons.css">
    
    
      <link rel="stylesheet" href="../../../../css/extra.css">
    
    
      
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#genome" tabindex="0" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="../../../.." title="St. Jude Cloud Documentation" aria-label="St. Jude Cloud Documentation" class="md-header-nav__button md-logo">
          
            <img alt="logo" src="../../../../images/stjude-logo-child-white.png" width="40" height="28">
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            <span class="md-header-nav__topic">
              St. Jude Cloud Documentation
            </span>
            <span class="md-header-nav__topic">
              
                Organizing Data into a Study View
              
            </span>
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" aria-label="search" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  

<a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    stjudecloud/docs
  </div>
</a>
          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../../.." class="md-tabs__link">
        Overview
      </a>
    
  </li>

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../covid-19/" class="md-tabs__link">
          COVID-19 Quickstart
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../genomics-platform/" class="md-tabs__link">
          Genomics Platform
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../pecan/" class="md-tabs__link">
          PeCan
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../visualization-community/" class="md-tabs__link">
          Visualization Community
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../" class="md-tabs__link md-tabs__link--active">
          ProteinPaint
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../studies/" class="md-tabs__link">
          Studies
        </a>
      
    </li>
  

      
        
      
        
      
    </ul>
  </div>
</nav>
      
      <main class="md-main" role="main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <a href="../../../.." title="St. Jude Cloud Documentation" class="md-nav__button md-logo">
      
        <img alt="logo" src="../../../../images/stjude-logo-child-white.png" width="48" height="48">
      
    </a>
    St. Jude Cloud Documentation
  </label>
  
    <div class="md-nav__source">
      


  

<a href="https://github.com/stjudecloud/docs/" title="Go to repository" class="md-source" data-md-source="github">
  
    <div class="md-source__icon">
      <svg viewBox="0 0 24 24" width="24" height="24">
        <use xlink:href="#__github" width="24" height="24"></use>
      </svg>
    </div>
  
  <div class="md-source__repository">
    stjudecloud/docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../../.." title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../../ecosystem/" title="About our Ecosystem" class="md-nav__link">
      About our Ecosystem
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      COVID-19 Quickstart
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        COVID-19 Quickstart
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../covid-19/" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../covid-19/getting-started/" title="Getting started" class="md-nav__link">
      Getting started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../covid-19/upload-local/" title="Data upload (local)" class="md-nav__link">
      Data upload (local)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../covid-19/upload-cluster/" title="Data upload (HPC)" class="md-nav__link">
      Data upload (HPC)
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../covid-19/creating-a-cloud-app-covid/" title="Creating a cloud app" class="md-nav__link">
      Creating a cloud app
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../covid-19/interactive-node/" title="Interactive nodes" class="md-nav__link">
      Interactive nodes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../covid-19/visualizing-ngs-data/" title="Visualizing NGS data" class="md-nav__link">
      Visualizing NGS data
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Genomics Platform
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Genomics Platform
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/accounts-and-billing/" title="Accounts and Billing" class="md-nav__link">
      Accounts and Billing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3">
    
    <label class="md-nav__link" for="nav-4-3">
      Requesting Data
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-3">
        Requesting Data
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/requesting-data/about-our-data/" title="About our Data" class="md-nav__link">
      About our Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/requesting-data/data-request/" title="Making a Data Request" class="md-nav__link">
      Making a Data Request
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/requesting-data/how-to-fill-out-DAA/" title="Filling Out a Data Access Agreement" class="md-nav__link">
      Filling Out a Data Access Agreement
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/requesting-data/glossary/" title="About our Decision Process & Terminology" class="md-nav__link">
      About our Decision Process & Terminology
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-4" type="checkbox" id="nav-4-4">
    
    <label class="md-nav__link" for="nav-4-4">
      Managing Data
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-4">
        Managing Data
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/managing-data/working-with-our-data/" title="Managing Data Overview" class="md-nav__link">
      Managing Data Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/managing-data/data-transfer-app/" title="Uploading/Downloading Data" class="md-nav__link">
      Uploading/Downloading Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/managing-data/how-to-fill-out-Extension/" title="Renewing your Data Access" class="md-nav__link">
      Renewing your Data Access
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-5" type="checkbox" id="nav-4-5">
    
    <label class="md-nav__link" for="nav-4-5">
      Analyzing Data
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-5">
        Analyzing Data
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/command-line/" title="Command Line Interaction" class="md-nav__link">
      Command Line Interaction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/running-sj-workflows/" title="Running our Workflows" class="md-nav__link">
      Running our Workflows
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-5-3" type="checkbox" id="nav-4-5-3">
    
    <label class="md-nav__link" for="nav-4-5-3">
      SJ Cloud Workflows
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="3">
      <label class="md-nav__title" for="nav-4-5-3">
        SJ Cloud Workflows
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/neoepitope/" title="NeoepitopePred" class="md-nav__link">
      NeoepitopePred
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/chipseq/" title="ChIP-Seq Peak Calling" class="md-nav__link">
      ChIP-Seq Peak Calling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/rapid-rnaseq/" title="Rapid RNA-Seq Fusion Detection" class="md-nav__link">
      Rapid RNA-Seq Fusion Detection
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/warden/" title="WARDEN Differential Expression Analysis" class="md-nav__link">
      WARDEN Differential Expression Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/mutational-signatures/" title="Mutational Signatures" class="md-nav__link">
      Mutational Signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/sequencerr/" title="SequencErr" class="md-nav__link">
      SequencErr
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/cis-x/" title="cis-X" class="md-nav__link">
      cis-X
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/analyzing-data/creating-a-cloud-app/" title="Creating your own Workflow" class="md-nav__link">
      Creating your own Workflow
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../genomics-platform/faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      PeCan
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        PeCan
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../pecan/" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../pecan/pecan-pie/" title="Pecan PIE" class="md-nav__link">
      Pecan PIE
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Visualization Community
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Visualization Community
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/proteinpaint/" title="ProteinPaint" class="md-nav__link">
      ProteinPaint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/genomepaint/" title="GenomePaint" class="md-nav__link">
      GenomePaint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../visualization-community/faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7" checked>
    
    <label class="md-nav__link" for="nav-7">
      ProteinPaint
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        ProteinPaint
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../" title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-2" type="checkbox" id="nav-7-2">
    
    <label class="md-nav__link" for="nav-7-2">
      User Guide
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-7-2">
        User Guide
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../navigating-the-protein-diagram/" title="Getting Started" class="md-nav__link">
      Getting Started
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../loading-datasets/" title="Loading Datasets" class="md-nav__link">
      Loading Datasets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../view-mutation-details/" title="Viewing SNV/Indel Mutations" class="md-nav__link">
      Viewing SNV/Indel Mutations
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../exploring-gene-fusions/" title="Viewing Gene Fusions" class="md-nav__link">
      Viewing Gene Fusions
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../exploring-gene-expression/" title="Viewing Expression Data" class="md-nav__link">
      Viewing Expression Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../filtering-mutations/" title="Filtering Mutations" class="md-nav__link">
      Filtering Mutations
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../exporting-figures-and-data/" title="Exporting Figures and Data" class="md-nav__link">
      Exporting Figures and Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/using-the-gene-table/" title="Using the Gene Table" class="md-nav__link">
      Using the Gene Table
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/using-the-sample-table/" title="Using the Sample Table" class="md-nav__link">
      Using the Sample Table
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/using-the-heatmap/" title="Using the Heatmap" class="md-nav__link">
      Using the Heatmap
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/using-disease-pie-charts/" title="Using Disease Pie Charts" class="md-nav__link">
      Using Disease Pie Charts
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/stratify-mutation-by-cancer-subtype/" title="Stratifying Mutations by Cancer Subtype" class="md-nav__link">
      Stratifying Mutations by Cancer Subtype
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-3" type="checkbox" id="nav-7-3">
    
    <label class="md-nav__link" for="nav-7-3">
      File Formats
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-7-3">
        File Formats
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/snv-and-indel/" title="SNV and Indel" class="md-nav__link">
      SNV and Indel
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/sv-and-fusion/" title="SV and fusion transcript" class="md-nav__link">
      SV and fusion transcript
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/cnv/" title="Copy Number Variants" class="md-nav__link">
      Copy Number Variants
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/itd/" title="Internal tandem duplications" class="md-nav__link">
      Internal tandem duplications
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/intragenic-deletion/" title="Intragenic deletion" class="md-nav__link">
      Intragenic deletion
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/truncation/" title="Trunction, N-loss or C-loss" class="md-nav__link">
      Trunction, N-loss or C-loss
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/json-bed/" title="JSON-BED" class="md-nav__link">
      JSON-BED
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/splice-junction/" title="Splice Junction" class="md-nav__link">
      Splice Junction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/pgv/" title="Profile-Gene-Value" class="md-nav__link">
      Profile-Gene-Value
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../file-formats/bampile/" title="Bampile" class="md-nav__link">
      Bampile
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-4" type="checkbox" id="nav-7-4">
    
    <label class="md-nav__link" for="nav-7-4">
      Advanced Usage
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-7-4">
        Advanced Usage
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/custom-protein-domains/" title="Add Custom Protein Domains" class="md-nav__link">
      Add Custom Protein Domains
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/using-custom-tracks/" title="Using Custom Tracks" class="md-nav__link">
      Using Custom Tracks
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/visualize-custom-mutations/" title="Visualize Your Mutation Data" class="md-nav__link">
      Visualize Your Mutation Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/loading-snv-indel-mutations-from-a-file/" title="Loading SNV/Indel Mutations from a File" class="md-nav__link">
      Loading SNV/Indel Mutations from a File
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/uploaded-mutation-data-user-interface/" title="Uploaded Mutation Data User Interface" class="md-nav__link">
      Uploaded Mutation Data User Interface
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/using-a-vcf-file/" title="Loading a TCGA MAF File" class="md-nav__link">
      Loading a TCGA MAF File
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../advanced-guides/using-a-vcf-file/" title="Loading a VCF File" class="md-nav__link">
      Loading a VCF File
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7-5" type="checkbox" id="nav-7-5" checked>
    
    <label class="md-nav__link" for="nav-7-5">
      Developer's Guide
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-7-5">
        Developer's Guide
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../embedding-proteinpaint/" title="Embedding ProteinPaint" class="md-nav__link">
      Embedding ProteinPaint
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Organizing Data into a Study View
      </label>
    
    <a href="./" title="Organizing Data into a Study View" class="md-nav__link md-nav__link--active">
      Organizing Data into a Study View
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#genome" class="md-nav__link">
    genome:""
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#name" class="md-nav__link">
    name:""
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assays-tracks-and-genome-browser" class="md-nav__link">
    Assays, tracks, and genome browser
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assays" class="md-nav__link">
    "assays":[]
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bigwig-tracks" class="md-nav__link">
    bigWig tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#splice-junction-tracks" class="md-nav__link">
    Splice junction tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-tracks" class="md-nav__link">
    VCF tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assay-of-ai-allelic-imbalance-tracks" class="md-nav__link">
    Assay of AI (allelic imbalance) tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assay-of-positional-annotation-tracks" class="md-nav__link">
    Assay of positional annotation tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#facet-tables" class="md-nav__link">
    Facet tables
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#browserview" class="md-nav__link">
    "browserview":{}
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mutation-data-in-tabular-format" class="md-nav__link">
    Mutation data in tabular format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mutationset" class="md-nav__link">
    mutationset:[]
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dbexpression" class="md-nav__link">
    dbexpression:{}
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-sample-heatmap" class="md-nav__link">
    Gene-sample heatmap
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#heatmapjson" class="md-nav__link">
    heatmapJSON:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hardcodemap" class="md-nav__link">
    hardcodemap:[]
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metadata-annotation" class="md-nav__link">
    Metadata annotation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#patientannotation" class="md-nav__link">
    "patientannotation":{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotations" class="md-nav__link">
    "annotations":{}
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#idkey-string" class="md-nav__link">
    .idkey "STRING"
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#files-array" class="md-nav__link">
    .files [ARRAY]
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputformat-string" class="md-nav__link">
    inputFormat "STRING"
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#expression-pca-plot" class="md-nav__link">
    Expression - PCA plot
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#e2pca" class="md-nav__link">
    "e2pca":[]
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#other-controls" class="md-nav__link">
    Other controls
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sankey" class="md-nav__link">
    sankey:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#diseasecolor" class="md-nav__link">
    diseasecolor:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genenetwork" class="md-nav__link">
    genenetwork:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#no_defaultgeneset1" class="md-nav__link">
    no_defaultgeneset:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_genetable1" class="md-nav__link">
    show_genetable:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_sampletable1" class="md-nav__link">
    show_sampletable:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#disable_sampletable1" class="md-nav__link">
    disable_sampletable:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_heatmap1" class="md-nav__link">
    show_heatmap:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_browser1" class="md-nav__link">
    show_browser:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_hardcodemap1" class="md-nav__link">
    show_hardcodemap:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_e2pca1" class="md-nav__link">
    "show_e2pca":1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#disable_genenetwork1" class="md-nav__link">
    disable_genenetwork:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hide_addnewfile1" class="md-nav__link">
    hide_addnewfile:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hide_navigation1" class="md-nav__link">
    hide_navigation:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#individual_label_namepatient" class="md-nav__link">
    individual_label_name:"patient"
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#obsolete-contents" class="md-nav__link">
    Obsolete contents
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#attribute-aicheck-moved-to-assays" class="md-nav__link">
    Attribute: aicheck (moved to "assays")
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#numerical-track-y-scale-configuration" class="md-nav__link">
    Numerical track Y-scale configuration
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#file-or-url" class="md-nav__link">
    File or URL?
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-bedgraph" class="md-nav__link">
    Track: bedGraph
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-bigwig" class="md-nav__link">
    Track: bigWig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-numeric2-a-pair-of-numerical-tracks" class="md-nav__link">
    Track: numeric2 (a pair of numerical tracks)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-junction-rna-seq-junction-reads" class="md-nav__link">
    Track: junction (RNA-seq junction reads)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-vcf" class="md-nav__link">
    Track: VCF
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-vafs1-variant-allele-fraction-of-a-single-sample" class="md-nav__link">
    Track: vafs1 (variant allele fraction of a single sample)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-bam" class="md-nav__link">
    Track: BAM
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-8" type="checkbox" id="nav-8">
    
    <label class="md-nav__link" for="nav-8">
      Studies
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-8">
        Studies
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../studies/" title="About" class="md-nav__link">
      About
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../../studies/sickle-cell/" title="Sickle Cell" class="md-nav__link">
      Sickle Cell
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../../faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../../citing-stjude-cloud/" title="Citing St. Jude Cloud" class="md-nav__link">
      Citing St. Jude Cloud
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#genome" class="md-nav__link">
    genome:""
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#name" class="md-nav__link">
    name:""
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assays-tracks-and-genome-browser" class="md-nav__link">
    Assays, tracks, and genome browser
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#assays" class="md-nav__link">
    "assays":[]
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bigwig-tracks" class="md-nav__link">
    bigWig tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#splice-junction-tracks" class="md-nav__link">
    Splice junction tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-tracks" class="md-nav__link">
    VCF tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assay-of-ai-allelic-imbalance-tracks" class="md-nav__link">
    Assay of AI (allelic imbalance) tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assay-of-positional-annotation-tracks" class="md-nav__link">
    Assay of positional annotation tracks
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#facet-tables" class="md-nav__link">
    Facet tables
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#browserview" class="md-nav__link">
    "browserview":{}
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mutation-data-in-tabular-format" class="md-nav__link">
    Mutation data in tabular format
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mutationset" class="md-nav__link">
    mutationset:[]
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#dbexpression" class="md-nav__link">
    dbexpression:{}
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gene-sample-heatmap" class="md-nav__link">
    Gene-sample heatmap
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#heatmapjson" class="md-nav__link">
    heatmapJSON:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hardcodemap" class="md-nav__link">
    hardcodemap:[]
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metadata-annotation" class="md-nav__link">
    Metadata annotation
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#patientannotation" class="md-nav__link">
    "patientannotation":{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#annotations" class="md-nav__link">
    "annotations":{}
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#idkey-string" class="md-nav__link">
    .idkey "STRING"
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#files-array" class="md-nav__link">
    .files [ARRAY]
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputformat-string" class="md-nav__link">
    inputFormat "STRING"
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#expression-pca-plot" class="md-nav__link">
    Expression - PCA plot
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#e2pca" class="md-nav__link">
    "e2pca":[]
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#other-controls" class="md-nav__link">
    Other controls
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sankey" class="md-nav__link">
    sankey:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#diseasecolor" class="md-nav__link">
    diseasecolor:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genenetwork" class="md-nav__link">
    genenetwork:{}
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#no_defaultgeneset1" class="md-nav__link">
    no_defaultgeneset:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_genetable1" class="md-nav__link">
    show_genetable:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_sampletable1" class="md-nav__link">
    show_sampletable:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#disable_sampletable1" class="md-nav__link">
    disable_sampletable:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_heatmap1" class="md-nav__link">
    show_heatmap:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_browser1" class="md-nav__link">
    show_browser:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_hardcodemap1" class="md-nav__link">
    show_hardcodemap:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#show_e2pca1" class="md-nav__link">
    "show_e2pca":1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#disable_genenetwork1" class="md-nav__link">
    disable_genenetwork:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hide_addnewfile1" class="md-nav__link">
    hide_addnewfile:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hide_navigation1" class="md-nav__link">
    hide_navigation:1
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#individual_label_namepatient" class="md-nav__link">
    individual_label_name:"patient"
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#obsolete-contents" class="md-nav__link">
    Obsolete contents
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#attribute-aicheck-moved-to-assays" class="md-nav__link">
    Attribute: aicheck (moved to "assays")
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#numerical-track-y-scale-configuration" class="md-nav__link">
    Numerical track Y-scale configuration
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#file-or-url" class="md-nav__link">
    File or URL?
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-bedgraph" class="md-nav__link">
    Track: bedGraph
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-bigwig" class="md-nav__link">
    Track: bigWig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-numeric2-a-pair-of-numerical-tracks" class="md-nav__link">
    Track: numeric2 (a pair of numerical tracks)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-junction-rna-seq-junction-reads" class="md-nav__link">
    Track: junction (RNA-seq junction reads)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-vcf" class="md-nav__link">
    Track: VCF
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-vafs1-variant-allele-fraction-of-a-single-sample" class="md-nav__link">
    Track: vafs1 (variant allele fraction of a single sample)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#track-bam" class="md-nav__link">
    Track: BAM
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/stjudecloud/docs/edit/master/docs/guides/proteinpaint/developers-guide/organizing-data-into-a-study-view.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                  <h1>Organizing Data into a Study View</h1>
                
                <p><img alt="" src="../../../images/guides/proteinpaint/developers-guide/organizing-data-into-a-study/media/image1.png" /></p>
<p>A nicely-organized study. The study is expressed as a JSON file, which contains one single object. To debug JSON, try <a href="http://jsonlint.com/">http://jsonlint.com/</a>.</p>
<p>Example study:</p>
<div class="highlight"><pre><span></span><code><span class="p">{</span>
    <span class="s2">&quot;genome&quot;</span><span class="o">:</span><span class="s2">&quot;hg19&quot;</span><span class="p">,</span>
    <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;example study&quot;</span><span class="p">,</span>
    <span class="s2">&quot;mutationset&quot;</span><span class="o">:</span><span class="p">[{</span>
        <span class="s2">&quot;snvindel&quot;</span><span class="o">:</span><span class="s2">&quot;path/to/file.txt&quot;</span><span class="p">,</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;DNA mutation&quot;</span>
    <span class="p">}],</span>
    <span class="s2">&quot;show_genetable&quot;</span><span class="o">:</span><span class="mi">1</span>
<span class="p">}</span>
</code></pre></div>

<p>Study attributes are listed below, they are case-sensitive.</p>
<h3 id="genome"><code>genome:""</code><a class="headerlink" href="#genome" title="Permanent link">&para;</a></h3>
<p>Name of the reference genome, e.g. hg19 or hg38</p>
<h3 id="name"><code>name:""</code><a class="headerlink" href="#name" title="Permanent link">&para;</a></h3>
<p>The name of this study/cohort.</p>
<p>Assay, tracks, and genome browser</p>
<h2 id="assays-tracks-and-genome-browser">Assays, tracks, and genome browser<a class="headerlink" href="#assays-tracks-and-genome-browser" title="Permanent link">&para;</a></h2>
<h3 id="assays"><code>"assays":[]</code><a class="headerlink" href="#assays" title="Permanent link">&para;</a></h3>
<p>Declares all assay names. Example:</p>
<div class="highlight"><pre><span></span><code><span class="p">{</span>
   <span class="s2">&quot;assays&quot;</span><span class="o">:</span><span class="p">[</span><span class="s2">&quot;H3K4me3&quot;</span><span class="p">,</span><span class="s2">&quot;RNA-seq&quot;</span><span class="p">,</span><span class="s2">&quot;Splice junctions&quot;</span><span class="p">],</span>
   <span class="s2">&quot;H3K4me3&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="p">...</span> <span class="p">...</span> 
   <span class="p">},</span>
   <span class="s2">&quot;RNA-seq&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="p">...</span> <span class="p">...</span> 
   <span class="p">},</span>
   <span class="s2">&quot;Splice junctions&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="p">...</span> <span class="p">...</span> 
   <span class="p">}</span>
<span class="p">}</span>
</code></pre></div>

<ul>
<li>
<p>Assay names can be arbitrary words e.g.</p>
<ul>
<li>
<p>"H3K4me3"</p>
</li>
<li>
<p>"RNA-seq"</p>
</li>
<li>
<p>"Splice junctions"</p>
</li>
<li>
<p>Do not use reserved words such as "assay" or "genome"</p>
</li>
</ul>
</li>
<li>
<p>Assay names are case-sensitive</p>
</li>
<li>
<p>Each name should appear as an attribute in the study.</p>
</li>
<li>
<p>Each assay is a set of tracks, organized by "individual"  "sample type"  track list</p>
<ul>
<li>
<p>There can be one or more "sample type" for an "individual"</p>
</li>
<li>
<p>In case this is not applicable e.g. for cell lines, <strong>use cell line name for both individual and sample type</strong> (the hierarchy is hardcoded...)</p>
</li>
<li>
<p>For each sample type, <strong>there can be one or multiple tracks associated</strong> with it</p>
<ul>
<li>
<p>If one track, the value is an object</p>
</li>
<li>
<p>If multiple tracks, the value is an array of track objects</p>
</li>
</ul>
</li>
</ul>
</li>
<li>
<p>Tracks in one assay should be the same type. Supported types:</p>
<ul>
<li>
<p>bigWig</p>
</li>
<li>
<p>Splice junction</p>
</li>
<li>
<p>VCF</p>
</li>
<li>
<p>Allelic imbalance (todo)</p>
</li>
</ul>
</li>
<li>
<p>Assay tracks can be presented in a sample-by-assay matrix, so called "facet table". The facet table can be generated by default, and can be customized. See section on facet table.</p>
</li>
</ul>
<h3 id="bigwig-tracks">bigWig tracks<a class="headerlink" href="#bigwig-tracks" title="Permanent link">&para;</a></h3>
<p>Declare an assay type for the bigWig track, e.g.:</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;RNA-seq&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;config&quot;</span><span class="o">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="o">:</span> <span class="s2">&quot;bigwig&quot;</span><span class="p">,</span>
            <span class="s2">&quot;pcolor&quot;</span><span class="o">:</span> <span class="s2">&quot;red&quot;</span><span class="p">,</span>
            <span class="s2">&quot;height&quot;</span><span class="o">:</span> <span class="mi">70</span><span class="p">,</span>
            <span class="s2">&quot;scale&quot;</span><span class="o">:</span><span class="p">{</span>   <span class="s2">&quot;max&quot;</span><span class="o">:</span><span class="mi">500</span><span class="p">,</span>   <span class="s2">&quot;min&quot;</span><span class="o">:</span><span class="mi">0</span>   <span class="p">}</span>
    <span class="p">},</span>
    <span class="s2">&quot;patient1&quot;</span><span class="o">:</span> <span class="p">{</span>
            <span class="s2">&quot;sampletype1&quot;</span><span class="o">:</span> <span class="p">{</span>
                     <span class="s2">&quot;file&quot;</span><span class="o">:</span> <span class="s2">&quot;path/to/RNA-patient1-sampletype1.bw&quot;</span><span class="p">,</span>
                     <span class="s2">&quot;pcolor&quot;</span><span class="o">:</span><span class="s2">&quot;green&quot;</span>
            <span class="p">},</span>
            <span class="s2">&quot;sampletype2&quot;</span><span class="o">:</span> <span class="p">[</span>
                 <span class="p">...</span> <span class="nx">a</span> <span class="nx">list</span> <span class="k">of</span> <span class="nx">tracks</span> <span class="p">...</span> 
            <span class="p">],</span>
            <span class="p">....</span> <span class="nx">more</span> <span class="nx">sample</span> <span class="nx">types</span>
      <span class="p">},</span>
    <span class="p">...</span> <span class="nx">more</span> <span class="nx">tracks</span>
<span class="p">},</span>
</code></pre></div>

<ul>
<li>
<p><code>config : { }</code></p>
<ul>
<li>
<p><code>config.type</code> value must be "bigwig"</p>
</li>
<li>
<p>The rest of config contains type-specific configurations that will be applied to every track in this assay. See below link for all available options.</p>
</li>
</ul>
</li>
<li>
<p>[ Patient name ]</p>
<ul>
<li>
<p>Name of the patient/individual, value is a hash of types of samples from this patient</p>
</li>
<li>
<p>Note that the 2-level patient-sampletype arrangement is hard-coded and applies to any type of assay.</p>
</li>
</ul>
</li>
<li>
<p>[ Sample type ]</p>
<ul>
<li>
<p>Value could be an object or an array of objects</p>
<ul>
<li>Each object is a track</li>
</ul>
</li>
<li>
<p>The track object should contain "file" or "url" to point to the source of the track file</p>
</li>
<li>
<p>The track object can contain any configurations and will override the ones in "config".</p>
</li>
</ul>
</li>
</ul>
<p><a href="https://docs.google.com/document/d/1ZnPZKSSajWyNISSLELMozKxrZHQbdxQkkkQFnxw6zTs/edit#heading=h.6spyog171fm9">Read bigWig track configuration options.</a></p>
<h3 id="splice-junction-tracks">Splice junction tracks<a class="headerlink" href="#splice-junction-tracks" title="Permanent link">&para;</a></h3>
<p>Declare an assay type for the junction tracks, e.g.:</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;junction&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;config&quot;</span><span class="o">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="o">:</span> <span class="s2">&quot;junction&quot;</span><span class="p">,</span>
            <span class="s2">&quot;categories&quot;</span><span class="o">:</span><span class="p">{</span>
                 <span class="s2">&quot;known&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;#9C9C9C&quot;</span><span class="p">,</span> <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Known&quot;</span><span class="p">},</span>
                 <span class="s2">&quot;novel&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;#cc0000&quot;</span><span class="p">,</span> <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Novel&quot;</span><span class="p">}</span>
             <span class="p">}</span>
    <span class="p">},</span>
    <span class="s2">&quot;patient1&quot;</span><span class="o">:</span> <span class="p">{</span>
            <span class="s2">&quot;sampletype1&quot;</span><span class="o">:</span> <span class="p">{</span>
                 <span class="s2">&quot;file&quot;</span><span class="o">:</span> <span class="s2">&quot;path/to/RNA-patient1-sampletype1.gz&quot;</span>
            <span class="p">},</span>
            <span class="p">....</span> <span class="nx">more</span> <span class="nx">sample</span> <span class="nx">types</span>
      <span class="p">},</span>
    <span class="p">...</span> <span class="nx">more</span> <span class="nx">tracks</span>
<span class="p">},</span>
</code></pre></div>

<ul>
<li>
<p><code>Config.type</code> value is "junction"</p>
</li>
<li>
<p>Config.categories lists types of junctions and their rendering colors</p>
</li>
</ul>
<p><a href="https://docs.google.com/document/d/1ZnPZKSSajWyNISSLELMozKxrZHQbdxQkkkQFnxw6zTs/edit#heading=h.ngg6s97cbli9">Read junction track configuration options.</a></p>
<p><a href="https://docs.google.com/document/d/1PFva3Mn-U4VPNW0vHHC-CSnYBeotRnqbhCMQPmyLQG4/edit?usp=sharing">Read junction file format.</a></p>
<h3 id="vcf-tracks">VCF tracks<a class="headerlink" href="#vcf-tracks" title="Permanent link">&para;</a></h3>
<p>Declare an assay type for the VCF tracks, e.g.:</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;VCF&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;config&quot;</span><span class="o">:</span> <span class="p">{</span>
            <span class="s2">&quot;type&quot;</span><span class="o">:</span> <span class="s2">&quot;vcf&quot;</span>
    <span class="p">},</span>
    <span class="s2">&quot;patient1&quot;</span><span class="o">:</span> <span class="p">{</span>
            <span class="s2">&quot;sampletype1&quot;</span><span class="o">:</span> <span class="p">{</span><span class="s2">&quot;file&quot;</span><span class="o">:</span> <span class="s2">&quot;path/to/patient1-sampletype1.vcf.gz&quot;</span><span class="p">},</span>
            <span class="p">....</span> <span class="nx">more</span> <span class="nx">sample</span> <span class="nx">types</span>
      <span class="p">},</span>
    <span class="p">...</span> <span class="nx">more</span> <span class="nx">tracks</span>
<span class="p">},</span>
</code></pre></div>

<p><a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">VCF format specification</a></p>
<p><a href="https://docs.google.com/document/d/1dbuYeQR6cgkpzcPaIChRFtXwolJVheoTr9NEW_Mfthw/edit?usp=sharing">Additional requirements on the VCF format in ProteinPaint</a></p>
<h3 id="assay-of-ai-allelic-imbalance-tracks">Assay of AI (allelic imbalance) tracks<a class="headerlink" href="#assay-of-ai-allelic-imbalance-tracks" title="Permanent link">&para;</a></h3>
<div class="admonition todo">
<p class="admonition-title">Todo</p>
</div>
<h3 id="assay-of-positional-annotation-tracks">Assay of positional annotation tracks<a class="headerlink" href="#assay-of-positional-annotation-tracks" title="Permanent link">&para;</a></h3>
<div class="admonition todo">
<p class="admonition-title">Todo</p>
<p>In JSON-BED format</p>
<p>E.g. called peaks ...</p>
</div>
<h3 id="facet-tables">Facet tables<a class="headerlink" href="#facet-tables" title="Permanent link">&para;</a></h3>
<p>To define what samples and assays are to be used in making a
facet table, use the "trackfacets" attribute.</p>
<p>The value is an array, where each element defines one facet table. This
way it allows multiple facet tables to be defined.</p>
<div class="highlight"><pre><span></span><code><span class="p">{</span>
 <span class="s2">&quot;trackfacets&quot;</span><span class="o">:</span><span class="p">[</span>
    <span class="p">{</span>
       <span class="s2">&quot;name&quot;</span><span class="o">:</span> <span class="s2">&quot;name of this table&quot;</span><span class="p">,</span>
       <span class="s2">&quot;samples&quot;</span><span class="o">:</span> <span class="p">[</span>  <span class="s2">&quot;sample1&quot;</span><span class="p">,</span> <span class="s2">&quot;sample2&quot;</span><span class="p">,</span> <span class="p">],</span>
       <span class="s2">&quot;assays&quot;</span><span class="o">:</span> <span class="p">[</span> <span class="s2">&quot;assay1&quot;</span><span class="p">,</span> <span class="s2">&quot;assay2&quot;</span> <span class="p">],</span>
       <span class="s2">&quot;nosortassay&quot;</span><span class="o">:</span><span class="mi">1</span><span class="p">,</span>
       <span class="s2">&quot;nosortsample&quot;</span><span class="o">:</span><span class="mi">1</span>
   <span class="p">},</span>

   <span class="p">...</span> <span class="nx">additional</span> <span class="nx">facet</span> <span class="nx">tables</span> <span class="p">...</span> 
 <span class="p">]</span>
<span class="p">}</span>
</code></pre></div>

<div class="admonition todo">
<p class="admonition-title">Todo</p>
</div>
<h3 id="browserview"><code>"browserview":{}</code><a class="headerlink" href="#browserview" title="Permanent link">&para;</a></h3>
<p>Will launch a browser and show some tracks over the given position in
the genome.</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;browserview&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;position&quot;</span><span class="o">:</span><span class="p">{</span> <span class="s2">&quot;chr&quot;</span><span class="o">:</span><span class="s2">&quot;chr10&quot;</span><span class="p">,</span> <span class="s2">&quot;start&quot;</span><span class="o">:</span><span class="mi">8081012</span><span class="p">,</span>  <span class="s2">&quot;stop&quot;</span><span class="o">:</span><span class="mi">8103311</span> <span class="p">},</span>
    <span class="s2">&quot;nativetracks&quot;</span><span class="o">:</span><span class="s2">&quot;refgene,repeatmasker&quot;</span><span class="p">,</span>
    <span class="s2">&quot;assays&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;my_rnaseq_assay&quot;</span><span class="o">:</span><span class="mi">1</span><span class="p">,</span>
        <span class="s2">&quot;my_splicejunction_assay&quot;</span><span class="o">:</span><span class="p">{</span>
             <span class="s2">&quot;combined&quot;</span><span class="o">:</span><span class="mi">1</span>
        <span class="p">}</span>
    <span class="p">},</span>
    <span class="s2">&quot;defaultassaytracks&quot;</span><span class="o">:</span><span class="p">[</span>
        <span class="p">{</span> <span class="s2">&quot;assay&quot;</span><span class="o">:</span><span class="s2">&quot;assayname1&quot;</span><span class="p">,</span> <span class="s2">&quot;level1&quot;</span><span class="o">:</span><span class="s2">&quot;patientname&quot;</span> <span class="p">},</span>
        <span class="p">{</span> <span class="p">...</span> <span class="nx">another</span> <span class="nx">track</span> <span class="nx">finder</span> <span class="p">...</span> <span class="p">}</span>
    <span class="p">],</span>
    <span class="s2">&quot;tracks&quot;</span><span class="o">:</span><span class="p">[</span>
         <span class="p">{</span> <span class="p">...</span> <span class="nx">track</span> <span class="mi">1</span> <span class="p">...</span> <span class="p">},</span>
         <span class="p">{</span> <span class="p">...</span> <span class="nx">track</span> <span class="mi">2</span> <span class="p">...</span> <span class="p">}</span>
     <span class="p">]</span>
<span class="p">}</span>
</code></pre></div>

<ul>
<li>
<p><code>browserview.position</code></p>
<ul>
<li>
<p>Optional. If not provided, will use default position of the genome.</p>
</li>
<li>
<p>Value can be string or object:</p>
<ul>
<li>
<p>{ <strong>chr</strong>:"...", <strong>start</strong>:..., <strong>stop</strong>:...}</p>
</li>
<li>
<p>"chr1:12345-67890"</p>
</li>
</ul>
</li>
<li>
<p>Coordinates are 0-based</p>
</li>
</ul>
</li>
<li>
<p><code>browserview.nativetracks</code></p>
<ul>
<li>
<p>Specify what "native" tracks to be loaded by default, usually the gene track. Check with your ProteinPaint server to find out what tracks are available "natively" for that genome.</p>
</li>
<li>
<p>Value can be:</p>
<ul>
<li>
<p>A string of track name</p>
</li>
<li>
<p>A string of multiple track names, joined by comma, no space</p>
</li>
<li>
<p>An array of track names</p>
</li>
</ul>
</li>
<li>
<p>Track names should match with what's specified for the tracks of that genome, but case-insensitive</p>
</li>
<li>
<p>TODO: allow customizations</p>
</li>
</ul>
</li>
<li>
<p><code>browserview.assays : { }</code></p>
<ul>
<li>
<p>Optional. To show all tracks from specified assays.</p>
</li>
<li>
<p>If you don't want to show all tracks of an assay, use "assaytracks" attributes instead (see below)</p>
</li>
<li>
<p>Value is an object. Keys are assay names, and the name must exist in the "assays" array.</p>
</li>
<li>
<p>Value to each assay can vary. In above example:</p>
<ul>
<li>
<p>By using value "1", all tracks from "rna_seq" will be displayed as separate tracks.</p>
</li>
<li>
<p>By using value {"<strong>combined</strong>":1}, all tracks from this assay type will be aggregated into one track. This only applies to these track types:</p>
<ul>
<li>
<p>Junction</p>
</li>
<li>
<p>VCF</p>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>
<p><code>browserview.defaultassaytracks : [ {} ]</code></p>
<ul>
<li>
<p>Optional. To select a set of tracks from the assays defined in this study, and show them by default.</p>
</li>
<li>
<p>Each element is a "track finder", with these attributes to pinpoint the track:</p>
<ul>
<li>
<p>"<strong>assay</strong>":""</p>
<ul>
<li>The assay name, required.</li>
</ul>
</li>
<li>
<p>"<strong>level1</strong>":""</p>
<ul>
<li>Patient name, but here it's a generic name. Required.</li>
</ul>
</li>
<li>
<p>"<strong>level2</strong>":""</p>
<ul>
<li>Optional, the sample type. If specified, will use track from only this sample type of the patient as specified by "level1". Else, it will use all tracks from this patient.</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li>
<p><code>browserview.tracks : [ ]</code></p>
<ul>
<li>
<p>Optional. To provide a list of custom tracks.</p>
</li>
<li>
<p>Custom tracks defined as JSON objects, <a href="https://docs.google.com/document/d/1ZnPZKSSajWyNISSLELMozKxrZHQbdxQkkkQFnxw6zTs/edit?usp=sharing">see JSON format of tracks</a>.</p>
</li>
<li>
<p>This allows adding tracks in freedom, directly encoded, rather than systematic way of "assays". Such tracks are not associated with any assay type and won't show up in facet table.</p>
</li>
</ul>
</li>
</ul>
<h2 id="mutation-data-in-tabular-format">Mutation data in tabular format<a class="headerlink" href="#mutation-data-in-tabular-format" title="Permanent link">&para;</a></h2>
<h3 id="mutationset"><code>mutationset:[]</code><a class="headerlink" href="#mutationset" title="Permanent link">&para;</a></h3>
<p>Lists all mutation datasets for this cohort, and the mutation data files associated for each set.</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;mutationset&quot;</span><span class="o">:</span><span class="p">[</span>
     <span class="p">{</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;DNA somatic&quot;</span><span class="p">,</span>
        <span class="s2">&quot;snvindel&quot;</span><span class="o">:</span><span class="s2">&quot;datadir/snvindel_dna&quot;</span><span class="p">,</span>
        <span class="s2">&quot;sv&quot;</span><span class="o">:</span><span class="s2">&quot;datadir/sv&quot;</span><span class="p">,</span>
        <span class="s2">&quot;cnv&quot;</span><span class="o">:</span><span class="s2">&quot;datadir/cnv&quot;</span>
     <span class="p">},</span>
     <span class="p">{</span>
         <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;RNA-seq&quot;</span><span class="p">,</span>
         <span class="s2">&quot;snvindel&quot;</span><span class="o">:</span><span class="s2">&quot;datadir/snvindel_rna&quot;</span><span class="p">,</span>
         <span class="s2">&quot;fusion&quot;</span><span class="o">:</span><span class="s2">&quot;datadir/fusion&quot;</span>
     <span class="p">}</span>
<span class="p">],</span>
</code></pre></div>

<p>This example provides two sets of mutation for a cohort from both DNA
and RNA, as indicated in the name. The name can be arbitrary string. In
each dataset, different types of mutations are provided in the form of
separate text files.</p>
<p>The "datadir" from above example is relative to the ROOT directory.</p>
<p>Mutation data type keys, each should point to a server-hosted text file.</p>
<ul>
<li>
<p><strong>snvindel</strong></p>
<ul>
<li>SNV/indel data, <a href="https://drive.google.com/open?id=1OJ9aXq2_-a3BfIQdKLYCYzrJRTpu4_9i3gephTY-Z38">tabular text format</a></li>
</ul>
</li>
<li>
<p><strong>sv</strong></p>
</li>
<li>
<p><strong>fusion</strong></p>
<ul>
<li>
<p>Structural variation OR fusion, they are in the same <a href="https://plus.google.com/+XinZhou_s/posts/9FknbtqWKgZ">tabular text format</a></p>
</li>
<li>
<p>Note that "sv" is used in "DNA", and "fusion" is used in "RNA-seq"</p>
</li>
</ul>
</li>
<li>
<p><strong>svfusion</strong></p>
<ul>
<li>JSON format data for sv/fusion/ITD/deletion/truncation, format is the same as the JSON file made by fusion editor export.</li>
</ul>
</li>
<li>
<p><strong>cnv</strong></p>
<ul>
<li>Copy number change, <a href="https://plus.google.com/+XinZhou_s/posts/AqkeGvyxnN9">tabular text format</a></li>
</ul>
</li>
<li>
<p><strong>deletion</strong></p>
<ul>
<li>Intragenic deletion, <a href="https://drive.google.com/open?id=1tWbf3rg3BmVIZPGGPk023P0aBkDw_ry5XuZLGyGodyg">tabular text format</a></li>
</ul>
</li>
<li>
<p><strong>truncation</strong></p>
<ul>
<li>Gene truncation, <a href="https://drive.google.com/open?id=1P1g-Y8r30pSKfan1BhYZcsUtSk7wRb4plaO1S-JCJr4">tabular text format</a></li>
</ul>
</li>
<li>
<p><strong>Variant_gene_assoc</strong></p>
<ul>
<li>
<p><strong>No longer supported!!</strong></p>
</li>
<li>
<p>For describing genes associated with particular variants</p>
</li>
<li>
<p>Per-sample based: each line is one variant from one sample.</p>
</li>
<li>
<p>This will trigger a browser view for exploring variant and genes together.</p>
</li>
<li>
<p>7 columns:</p>
<p>i.  Chromosome</p>
<p>ii. Position</p>
<p>iii. Reference_allele</p>
<p>iv. Mutant_allele</p>
<p>v.  Patient</p>
<p>vi. Sampletype</p>
<p>vii. <strong>geneset</strong></p>
</li>
<li>
<p>The "geneset" is a JSON array, representing associated genes from each sample, example:</p>
<p>i.  [{"name":"TAL1","isoform":"NM_003189","position":"chr1:47681961-47698007","score_1":0.543,"score_2":0.543}, {... second gene ...} ]</p>
</li>
</ul>
</li>
</ul>
<h3 id="dbexpression"><code>dbexpression:{}</code><a class="headerlink" href="#dbexpression" title="Permanent link">&para;</a></h3>
<p>It provides a gene-level expression values over a group of samples from
a database table on the server, and show it when the gene is shown in
the protein-view.</p>
<p>"Dbexpression" should work with "mutationset".</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;dbexpression&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;dbfile&quot;</span><span class="o">:</span><span class="s2">&quot;path/to/data.db&quot;</span><span class="p">,</span>
    <span class="s2">&quot;tablename&quot;</span><span class="o">:</span><span class="s2">&quot;[name of the table]&quot;</span><span class="p">,</span>
    <span class="s2">&quot;keyname&quot;</span><span class="o">:</span><span class="s2">&quot;[name of the table field to query against, e.g. gene]&quot;</span><span class="p">,</span>
    <span class="s2">&quot;tidy&quot;</span><span class="o">:</span><span class="s2">&quot;function(rows){rows.forEach(function(row){ ... })}&quot;</span><span class="p">,</span>
    <span class="s2">&quot;config&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;RNA-seq&quot;</span><span class="p">,</span>
        <span class="s2">&quot;sampletype&quot;</span><span class="o">:</span><span class="s2">&quot;sample&quot;</span><span class="p">,</span>
        <span class="s2">&quot;datatype&quot;</span><span class="o">:</span><span class="s2">&quot;FPKM&quot;</span><span class="p">,</span>
        <span class="s2">&quot;ongene&quot;</span><span class="o">:</span><span class="kc">true</span><span class="p">,</span>
        <span class="s2">&quot;hlcolor&quot;</span><span class="o">:</span><span class="s2">&quot;#f53d00&quot;</span><span class="p">,</span>
        <span class="s2">&quot;hlcolor2&quot;</span><span class="o">:</span><span class="s2">&quot;#FFBEA8&quot;</span><span class="p">,</span>
        <span class="s2">&quot;attrlst&quot;</span><span class="o">:</span><span class="p">[{</span><span class="s2">&quot;k&quot;</span><span class="o">:</span><span class="s2">&quot;patient&quot;</span><span class="p">},{</span><span class="s2">&quot;k&quot;</span><span class="o">:</span><span class="s2">&quot;sampletype&quot;</span><span class="p">}],</span>
        <span class="s2">&quot;cohort&quot;</span><span class="o">:</span><span class="p">{</span>
           <span class="s2">&quot;levels&quot;</span><span class="o">:</span><span class="p">[</span>
                <span class="p">{</span><span class="s2">&quot;k&quot;</span><span class="o">:</span><span class="s2">&quot;disease&quot;</span><span class="p">,</span><span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Disease&quot;</span><span class="p">}</span>
            <span class="p">]</span>
           <span class="p">}</span>
    <span class="p">}</span>
<span class="p">},</span>
</code></pre></div>

<ul>
<li>
<p>dbexpression.dbfile</p>
<ul>
<li>Path to the SQLite database file. Must be relative to the <TP> directory.</li>
</ul>
</li>
<li>
<p>dbexpression.tablename</p>
<ul>
<li>Name of the database table to query against with</li>
</ul>
</li>
<li>
<p>dbexpression.tidy</p>
<ul>
<li>Provides a Javascript function in the form of a string, argument is an array of all records fetched from the database, the function will apply certain tidying operation on each of the record</li>
</ul>
</li>
<li>
<p>dbexpression.attrlst</p>
<ul>
<li></li>
</ul>
</li>
</ul>
<p>Note that the db table can contain data for samples that are not from
this cohort, and they will still be displayed in the expression data
panel.</p>
<h2 id="gene-sample-heatmap">Gene-sample heatmap<a class="headerlink" href="#gene-sample-heatmap" title="Permanent link">&para;</a></h2>
<h3 id="heatmapjson"><code>heatmapJSON:{}</code><a class="headerlink" href="#heatmapjson" title="Permanent link">&para;</a></h3>
<p>An JSON object that configures the heatmap layout. <a href="https://docs.google.com/document/d/1bx3VRdTzJDYdmt4CaU4PfHCOz5JSfTyuSFIuGTx-X2g/edit">Refer to heatmap configuration tutorial.</a></p>
<h3 id="hardcodemap"><code>hardcodemap:[]</code><a class="headerlink" href="#hardcodemap" title="Permanent link">&para;</a></h3>
<p>Provides one or more "hard-coded heatmap".</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;hardcodemap&quot;</span><span class="o">:</span><span class="p">[</span>
    <span class="p">{</span>
    <span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;path/to/file.txt&quot;</span><span class="p">,</span>
    <span class="s2">&quot;rowh&quot;</span><span class="o">:</span><span class="mi">18</span><span class="p">,</span>
    <span class="s2">&quot;rowspace&quot;</span><span class="o">:</span><span class="mi">1</span><span class="p">,</span>
    <span class="s2">&quot;colw&quot;</span><span class="o">:</span><span class="mi">10</span><span class="p">,</span>
    <span class="s2">&quot;colspace&quot;</span><span class="o">:</span><span class="mi">1</span><span class="p">,</span>
    <span class="s2">&quot;metadata&quot;</span><span class="o">:</span><span class="p">{</span>
       <span class="s2">&quot;key1&quot;</span><span class="o">:</span><span class="p">{</span>
            <span class="s2">&quot;v1&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;what&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span><span class="p">},</span>
            <span class="s2">&quot;v2&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;what&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;blue&quot;</span><span class="p">}</span>
       <span class="p">},</span>
       <span class="s2">&quot;key2&quot;</span><span class="o">:</span><span class="p">{</span>
             <span class="s2">&quot;V3&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;what&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span><span class="p">},</span>
             <span class="s2">&quot;V4&quot;</span><span class="o">:</span><span class="s2">&quot;{&quot;</span><span class="nx">label</span><span class="s2">&quot;:&quot;</span><span class="nx">what</span><span class="s2">&quot;,&quot;</span><span class="nx">color</span><span class="s2">&quot;:&quot;</span><span class="nx">red</span><span class="err">&quot;</span><span class="p">}</span>
       <span class="p">}</span>
    <span class="p">}</span>
<span class="p">],</span>
</code></pre></div>

<p>File format:</p>
<ul>
<li>
<p>Columns are samples</p>
</li>
<li>
<p>Rows are items (e.g. genes, or other attributes describing samples)</p>
</li>
<li>
<p>First line is header</p>
<ul>
<li>
<p>First column is metadata keys, the value must be found as one of the keys in hardcodemap.metadata</p>
</li>
<li>
<p>Second column is row name</p>
</li>
</ul>
</li>
<li>
<p>Cell value is one of the values of each key (of that row). Join multiple values by semicolon.</p>
</li>
</ul>
<h2 id="metadata-annotation">Metadata annotation<a class="headerlink" href="#metadata-annotation" title="Permanent link">&para;</a></h2>
<p>(under development)</p>
<h3 id="patientannotation"><code>"patientannotation":{}</code><a class="headerlink" href="#patientannotation" title="Permanent link">&para;</a></h3>
<p>An object to define metadata annotation at patient level.</p>
<p>This should also be used when there is only information on samples but
not patients (in which case sample name/id should be expressed at
"patient" level).</p>
<p><strong>patientannotation.annotation</strong> provides actual annotation of each
patient/sample.</p>
<p><strong>patientannotation.metadata</strong> provides the list of metadata terms, and
attributes associated with the term.</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;patientannotation&quot;</span><span class="o">:</span><span class="p">{</span>

       <span class="s2">&quot;annotation&quot;</span><span class="o">:</span><span class="p">{</span>
               <span class="s2">&quot;patient1&quot;</span><span class="o">:</span><span class="p">{</span>
                     <span class="s2">&quot;age&quot;</span><span class="o">:</span><span class="s2">&quot;child&quot;</span><span class="p">,</span>
                     <span class="s2">&quot;sex&quot;</span><span class="o">:</span><span class="s2">&quot;m&quot;</span>
                <span class="p">},</span> 
                <span class="p">...</span> <span class="nx">more</span> <span class="nx">patients</span> <span class="p">...</span> 
        <span class="p">},</span>

<span class="s2">&quot;metadata&quot;</span><span class="o">:</span><span class="p">[</span>
   <span class="p">{</span>
   <span class="s2">&quot;key&quot;</span><span class="o">:</span><span class="s2">&quot;age&quot;</span><span class="p">,</span>
   <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Age at diagnosis&quot;</span><span class="p">,</span>
   <span class="s2">&quot;values&quot;</span><span class="o">:</span><span class="p">[</span>
      <span class="p">{</span>
      <span class="s2">&quot;key&quot;</span><span class="o">:</span><span class="s2">&quot;child&quot;</span><span class="p">,</span>
      <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Child (&lt;10yrs)&quot;</span><span class="p">,</span>
      <span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span>
      <span class="p">},</span>
      <span class="p">{</span>
      <span class="s2">&quot;key&quot;</span><span class="o">:</span><span class="s2">&quot;adult&quot;</span><span class="p">,</span>
      <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Adult (&gt;18yrs)&quot;</span><span class="p">,</span>
      <span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;blue&quot;</span>
      <span class="p">}</span>
   <span class="p">]</span>
   <span class="p">},</span>
   <span class="p">{</span>
   <span class="s2">&quot;key&quot;</span><span class="o">:</span><span class="s2">&quot;sex&quot;</span><span class="p">,</span>
   <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Sex&quot;</span><span class="p">,</span>
   <span class="s2">&quot;values&quot;</span><span class="o">:</span><span class="p">[</span>
      <span class="p">{</span>
      <span class="s2">&quot;key&quot;</span><span class="o">:</span><span class="s2">&quot;m&quot;</span><span class="p">,</span>
      <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Male&quot;</span><span class="p">,</span>
      <span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span>
      <span class="p">},</span>
      <span class="p">{</span>
      <span class="s2">&quot;key&quot;</span><span class="o">:</span><span class="s2">&quot;f&quot;</span><span class="p">,</span>
      <span class="s2">&quot;label&quot;</span><span class="o">:</span><span class="s2">&quot;Female&quot;</span><span class="p">,</span>
      <span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;blue&quot;</span>
      <span class="p">}</span>
   <span class="p">]</span>
   <span class="p">},</span>
   <span class="p">...</span> <span class="nx">more</span> <span class="nx">terms</span> <span class="p">...</span> 
<span class="p">]</span>
<span class="p">}</span>
</code></pre></div>

<p>Such metadata annotation can be applied to:</p>
<ul>
<li>
<p>Expression - PCA plot</p>
</li>
<li>
<p>Splice junction track (TODO)</p>
</li>
</ul>
<h3 id="annotations"><code>"annotations":{}</code><a class="headerlink" href="#annotations" title="Permanent link">&para;</a></h3>
<p>An object to define metadata annotation by input file.</p>
<h4 id="idkey-string"><code>.idkey "STRING"</code><a class="headerlink" href="#idkey-string" title="Permanent link">&para;</a></h4>
<p>The column value, typically "sample" or "samplename", to use for
collecting annotations into an object</p>
<h4 id="files-array"><code>.files [ARRAY]</code><a class="headerlink" href="#files-array" title="Permanent link">&para;</a></h4>
<p>An array of string filepaths to use as metadata source data</p>
<h4 id="inputformat-string"><code>inputFormat "STRING"</code><a class="headerlink" href="#inputformat-string" title="Permanent link">&para;</a></h4>
<p>Indicates the format for parsing the source file, defaults to
"metadataTsv"</p>
<h2 id="expression-pca-plot">Expression - PCA plot<a class="headerlink" href="#expression-pca-plot" title="Permanent link">&para;</a></h2>
<h3 id="e2pca"><code>"e2pca":[]</code><a class="headerlink" href="#e2pca" title="Permanent link">&para;</a></h3>
<p>A list of plots to be made, each defines an "expression - PCA" plot.</p>
<p><a href="https://docs.google.com/document/d/1Midt0rYs1iIJveUMjeng9si3q31YelcNe_PI3Njce9E/edit?usp=sharing">More on Expression PCA plot.</a></p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;e2pca&quot;</span><span class="o">:</span> <span class="p">[</span>
    <span class="p">{</span>
      <span class="s2">&quot;name&quot;</span><span class="o">:</span> <span class="s2">&quot;optional name of this plot&quot;</span><span class="p">,</span>
      <span class="s2">&quot;vectorfile&quot;</span><span class="o">:</span> <span class="s2">&quot;path/to/vectorfile.txt&quot;</span><span class="p">,</span>
      <span class="s2">&quot;dbfile&quot;</span><span class="o">:</span> <span class="s2">&quot;path/to/data.db&quot;</span><span class="p">,</span>
      <span class="s2">&quot;colorscale&quot;</span><span class="o">:</span><span class="p">{</span>
          <span class="s2">&quot;from&quot;</span><span class="o">:</span><span class="s2">&quot;blue&quot;</span><span class="p">,</span>
          <span class="s2">&quot;to&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span>
    <span class="p">}</span>
    <span class="p">},</span>
    <span class="p">...</span> <span class="nx">one</span> <span class="nx">more</span> <span class="nx">plot</span> <span class="p">...</span> 
<span class="p">],</span>
</code></pre></div>

<h2 id="other-controls">Other controls<a class="headerlink" href="#other-controls" title="Permanent link">&para;</a></h2>
<h3 id="sankey"><code>sankey:{}</code><a class="headerlink" href="#sankey" title="Permanent link">&para;</a></h3>
<p>An object of possible configurations to the disease-gene sankey diagram.</p>
<p>Use sankey.genes:[] to define a set of genes to be used in the
diagram:</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;sankey&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;genes&quot;</span><span class="o">:</span><span class="p">[</span>
        <span class="p">{</span><span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;ERG&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span><span class="p">},</span>
        <span class="p">{</span><span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;ZEB2&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span><span class="p">},</span>
        <span class="p">{</span><span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;MYC&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span><span class="p">},</span>
        <span class="p">{</span><span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;MYCBP2&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span><span class="p">},</span> <span class="p">....</span>
    <span class="p">]</span>
<span class="p">},</span>
</code></pre></div>

<p>Use sankey.geneset:[] to define gene sets to be used in the diagram,
all genes in a set share the same color, legend will be rendered:</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;sankey&quot;</span><span class="o">:</span><span class="p">{</span>
      <span class="s2">&quot;geneset&quot;</span><span class="o">:</span><span class="p">[</span>
            <span class="p">{</span><span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;gene set 1&quot;</span><span class="p">,</span>
             <span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;red&quot;</span><span class="p">,</span>
             <span class="s2">&quot;genes&quot;</span><span class="o">:</span><span class="p">[</span><span class="s2">&quot;ERG&quot;</span><span class="p">,</span><span class="s2">&quot;ZEB2&quot;</span><span class="p">]</span>
            <span class="p">},</span>
            <span class="p">{</span><span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;gene set 2&quot;</span><span class="p">,</span><span class="s2">&quot;color&quot;</span><span class="o">:</span><span class="s2">&quot;blue&quot;</span><span class="p">,</span><span class="s2">&quot;genes&quot;</span><span class="o">:</span><span class="p">[</span> <span class="p">......</span> <span class="p">]}</span>
       <span class="p">]</span>
<span class="p">},</span>
</code></pre></div>

<h3 id="diseasecolor"><code>diseasecolor:{}</code><a class="headerlink" href="#diseasecolor" title="Permanent link">&para;</a></h3>
<p>An object to define rendering color for different diseases. Applies to
sankey diagram and gene network chart.</p>
<h3 id="genenetwork"><code>genenetwork:{}</code><a class="headerlink" href="#genenetwork" title="Permanent link">&para;</a></h3>
<p>To provide one or more gene network diagrams.</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;genenetwork&quot;</span><span class="o">:</span><span class="p">{</span>
     <span class="s2">&quot;list&quot;</span><span class="o">:</span><span class="p">[</span> 
          <span class="p">{</span> <span class="nx">a</span> <span class="nx">JSON</span> <span class="nx">object</span> <span class="nx">representing</span> <span class="nx">a</span> <span class="nx">network</span> <span class="p">},</span>
          <span class="p">{</span> <span class="nx">one</span> <span class="nx">more</span> <span class="nx">network</span> <span class="p">}</span>
     <span class="p">]</span>
<span class="p">}</span>
</code></pre></div>

<p>Each network is defined as a JSON object, <a href="https://docs.google.com/document/d/1PAlu-bJMBuHRVpGhB_eVaOmlhhbWneC_94_7RRItuA8/edit?usp=sharing">see example</a>.</p>
<h3 id="no_defaultgeneset1"><code>no_defaultgeneset:1</code><a class="headerlink" href="#no_defaultgeneset1" title="Permanent link">&para;</a></h3>
<p>Will not apply the genome-equipped gene set to generate the heatmap
view, but rather, to use highly recurrent genes based on the data:</p>
<ul>
<li>Max 20 genes, all in one group.</li>
</ul>
<h3 id="show_genetable1"><code>show_genetable:1</code><a class="headerlink" href="#show_genetable1" title="Permanent link">&para;</a></h3>
<p>Will show the gene table by default. Only applicable when "mutationset"
is used. Value is "1" for true.</p>
<h3 id="show_sampletable1"><code>show_sampletable:1</code><a class="headerlink" href="#show_sampletable1" title="Permanent link">&para;</a></h3>
<p>Will show the sample table by default. Value is "1" for true.</p>
<p>Samples can come from either "mutationset", or assays.</p>
<h3 id="disable_sampletable1"><code>disable_sampletable:1</code><a class="headerlink" href="#disable_sampletable1" title="Permanent link">&para;</a></h3>
<p>Will hide the sample table.</p>
<h3 id="show_heatmap1"><code>show_heatmap:1</code><a class="headerlink" href="#show_heatmap1" title="Permanent link">&para;</a></h3>
<p>Will show the heatmap by default. Value is "1" for true.</p>
<h3 id="show_browser1"><code>show_browser:1</code><a class="headerlink" href="#show_browser1" title="Permanent link">&para;</a></h3>
<p>Will show the genome browser by default. Value is "1" for true.</p>
<h3 id="show_hardcodemap1"><code>show_hardcodemap:1</code><a class="headerlink" href="#show_hardcodemap1" title="Permanent link">&para;</a></h3>
<p>Will show the hard-coded heatmap by default.</p>
<h3 id="show_e2pca1"><code>"show_e2pca":1</code><a class="headerlink" href="#show_e2pca1" title="Permanent link">&para;</a></h3>
<p>Will show the expression - PCA plots by default.</p>
<h3 id="disable_genenetwork1"><code>disable_genenetwork:1</code><a class="headerlink" href="#disable_genenetwork1" title="Permanent link">&para;</a></h3>
<p>Will hide the gene network function.</p>
<h3 id="hide_addnewfile1"><code>hide_addnewfile:1</code><a class="headerlink" href="#hide_addnewfile1" title="Permanent link">&para;</a></h3>
<p>Will hide the "+NEW FILE" tab.</p>
<h3 id="hide_navigation1"><code>hide_navigation:1</code><a class="headerlink" href="#hide_navigation1" title="Permanent link">&para;</a></h3>
<p>Will hide all navigation buttons on the left.</p>
<h3 id="individual_label_namepatient"><code>individual_label_name:"patient"</code><a class="headerlink" href="#individual_label_namepatient" title="Permanent link">&para;</a></h3>
<p>Will replace the default "individual" with the given word in the "#
individuals" tab on left of the study view.</p>
<h2 id="obsolete-contents">Obsolete contents<a class="headerlink" href="#obsolete-contents" title="Permanent link">&para;</a></h2>
<p><span style='font-size: 100pt;'>STOP HERE.</span></p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>The rest of these contents are obsoletse. We're busy porting them to the new ProteinPaint.</p>
</div>
<h3 id="attribute-aicheck-moved-to-assays"><code>Attribute: aicheck (moved to "assays")</code><a class="headerlink" href="#attribute-aicheck-moved-to-assays" title="Permanent link">&para;</a></h3>
<p>This is the "aicheck" track.</p>
<p>This is the integrative display of allelic frequency imbalance of
variants or genetic markers, as well as DNA sequencing coverage over
these markers. Adapted from the "aicheck" figure invented by Xiaotu Ma.</p>
<p>Note that this is designed for tumor-normal matched paired samples only.</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;aicheck&quot;</span><span class="o">:</span> <span class="p">{</span>
    <span class="s2">&quot;20-PABLDZ&quot;</span><span class="o">:</span> <span class="p">{</span>
      <span class="s2">&quot;sampletypes&quot;</span><span class="o">:</span> <span class="p">[</span>
        <span class="s2">&quot;relapse&quot;</span><span class="p">,</span>
        <span class="s2">&quot;diagnosis&quot;</span>
      <span class="p">],</span>
      <span class="s2">&quot;relapse&quot;</span><span class="o">:</span> <span class="p">{</span>
        <span class="s2">&quot;file2&quot;</span><span class="o">:</span> <span class="s2">&quot;aml/ai_check_SJBALL013790_R1_G1.txt.gz&quot;</span><span class="p">,</span>
        <span class="s2">&quot;readdepthcutoff&quot;</span><span class="o">:</span> <span class="mi">126</span>
      <span class="p">},</span>
      <span class="s2">&quot;diagnosis&quot;</span><span class="o">:</span> <span class="p">{</span>
        <span class="s2">&quot;file&quot;</span><span class="o">:</span> <span class="s2">&quot;aml/fullMaf_TARGET-20-PABLDZ-09A-02D_NormalVsPrimary.maf.txt.gz&quot;</span><span class="p">,</span>
        <span class="s2">&quot;readdepthcutoff&quot;</span><span class="o">:</span> <span class="mi">126</span>
      <span class="p">}</span>
    <span class="p">},</span>
    <span class="s2">&quot;patient2&quot;</span><span class="o">:</span><span class="p">{</span> <span class="p">...</span> <span class="p">},</span>
    <span class="p">...</span> <span class="nx">more</span> <span class="nx">patients</span> <span class="p">...</span> 
<span class="p">},</span>
</code></pre></div>

<p>The "samplekeys" is an array of sample type. This is a relic of initial
design and will be abandoned. By then it will be ignored by
ProteinPaint.</p>
<p>There are two file formats for aicheck, identified by "file" and
"file2". "file2" format comes right off the CompBio pipeline (generated
by one of Xiaotu's script).</p>
<div class="admonition example">
<p class="admonition-title">"file" columns</p>
<ol>
<li>chromosome name, must be "chr1" but not "1"</li>
<li>coordinate, 1-based                        </li>
<li>MinD: mutant allele read count in tumor DNA</li>
<li>TinD: total read count in tumor DNA</li>
<li>MinN: mutant allele read count in normal DNA</li>
<li>TinN: total read count in normal DNA </li>
</ol>
</div>
<div class="admonition example">
<p class="admonition-title">"file2" columns</p>
<ol>
<li>chromosome name, must be "chr1" but not "1"</li>
<li>coordinate, 1-based</li>
<li>SNP, value seems always is "SNP"</li>
<li>rsNumber: dbSNP name</li>
<li>TinD: total read count in tumor DNA</li>
<li>dMAF: mutant allele frequency in tumor DNA</li>
<li>TinN: total read count in normal DNA</li>
<li>nMAF: mutant allele frequency in normal DNA</li>
</ol>
</div>
<p>The "readdepthcutoff" value defines the Y-axis max value of read depth
track for both tumor and normal. It is pre-calculated by following
Xiaotu's method below. User is free to use any other method.</p>
<div class="highlight"><pre><span></span><code><span class="c1">######## to calculate trimmed media, in 1% to 99% range</span>
<span class="n">trimed.median</span><span class="o">&lt;-</span><span class="nf">function</span><span class="p">(</span><span class="n">xx</span><span class="p">)</span> <span class="p">{</span>
  <span class="n">cutt</span><span class="o">&lt;-</span><span class="m">0</span>
  <span class="n">dnn</span><span class="o">&lt;-</span><span class="nf">quantile</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span><span class="m">0.01</span><span class="p">)</span>
  <span class="n">upp</span><span class="o">&lt;-</span><span class="nf">quantile</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span><span class="m">0.99</span><span class="p">)</span>
  <span class="nf">if</span><span class="p">(</span><span class="nf">sum</span><span class="p">(</span><span class="n">xx</span><span class="o">&gt;</span><span class="n">dnn</span> <span class="o">&amp;</span> <span class="n">xx</span><span class="o">&lt;</span><span class="n">upp</span><span class="p">)</span><span class="o">&gt;</span><span class="m">2</span><span class="p">)</span> <span class="p">{</span>
    <span class="n">use</span><span class="o">&lt;-</span><span class="p">(</span><span class="n">xx</span><span class="o">&gt;</span><span class="n">dnn</span> <span class="o">&amp;</span> <span class="n">xx</span><span class="o">&lt;</span><span class="n">upp</span><span class="p">)</span>
    <span class="n">cutt</span><span class="o">&lt;-</span><span class="nf">quantile</span><span class="p">(</span><span class="n">xx</span><span class="p">[</span><span class="n">use</span><span class="p">],</span><span class="m">0.98</span><span class="p">)</span>
  <span class="p">}</span>
  <span class="nf">return</span><span class="p">(</span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">cutt</span><span class="p">))</span>
<span class="p">}</span> 
<span class="c1">#Tumor</span>
  <span class="n">medd.D</span><span class="o">&lt;-</span><span class="nf">trimed.median</span><span class="p">(</span><span class="n">dt0</span><span class="p">[,</span><span class="s">&quot;TinD&quot;</span><span class="p">])</span><span class="o">*</span><span class="m">1.5</span>
<span class="c1">#Normal</span>
  <span class="n">medd.G</span><span class="o">&lt;-</span><span class="nf">trimed.median</span><span class="p">(</span><span class="n">dt0</span><span class="p">[,</span><span class="s">&quot;TinN&quot;</span><span class="p">])</span><span class="o">*</span><span class="m">1.5</span>
  <span class="nf">if</span><span class="p">(</span><span class="n">medd.D</span><span class="o">&lt;</span><span class="m">1</span><span class="p">)</span> <span class="n">medd.D</span><span class="o">&lt;-</span><span class="m">1</span>
  <span class="nf">if</span><span class="p">(</span><span class="n">medd.G</span><span class="o">&lt;</span><span class="m">1</span><span class="p">)</span> <span class="n">medd.G</span><span class="o">&lt;-</span><span class="m">1</span>
<span class="c1">#Maximum</span>
  <span class="n">medd</span><span class="o">&lt;-</span><span class="nf">max</span><span class="p">(</span><span class="n">medd.D</span><span class="p">,</span><span class="n">medd.G</span><span class="p">)</span>
</code></pre></div>

<p>Broadly, there are two types of tracks that ProteinPaint can read:
numerical, and positional. Bedgraph and bigWig are numerical track
formats, the generic bed is the positional format.</p>
<p>In the cohort definition, you have to first declare what the tracks are
about, then the type/format of the track. The first layer is assay type,
like RNA-seq, or it can be anything with a name. We will refer to it as
"assay type".</p>
<p>Assay types are predefined, allowed values are (case-sensitive):</p>
<ul>
<li>rnaseq</li>
</ul>
<p>Each assay type must be declared as a hash, with patient names as keys.
Patients from different assay types may or may not be the same.</p>
<p>There is always a reserved key called "config", providing configurations
applicable to all tracks of this assay type. Each track can have its own
configurations in order to be different from the "global" attributes in
config.</p>
<h3 id="numerical-track-y-scale-configuration">Numerical track Y-scale configuration<a class="headerlink" href="#numerical-track-y-scale-configuration" title="Permanent link">&para;</a></h3>
<ul>
<li>
<p><strong>percentile</strong></p>
<ul>
<li>
<p>Value is positive integer from 1 to 99</p>
</li>
<li>
<p>If provided, the nth percentile value in the view range will be used to set Y-axis. This is useful for preventing large outlier values from skewing the display.</p>
</li>
<li>
<p>When there are both positive and negative values in the view range, the percentile calculation will be applied separately to positive and negative values for calculating max and min values.</p>
</li>
</ul>
</li>
<li>
<p><strong>min, max</strong></p>
<ul>
<li>When valid values are provided for both, will set fixed scale.</li>
</ul>
</li>
<li>
<p><strong>autoscale</strong></p>
<ul>
<li>Apply auto-scale, will override percentile or min/max settings.</li>
</ul>
</li>
</ul>
<h3 id="file-or-url">File or URL?<a class="headerlink" href="#file-or-url" title="Permanent link">&para;</a></h3>
<ul>
<li>
<p><strong>url</strong></p>
<ul>
<li>
<p>HTTP or HTTPS, such as <a href="http://datahub.stjude.org/datasets/MBSE2016/hg19/tracks/1M1_H3K27Ac_treat_afterfiting_all.bw">http://datahub.stjude.org/datasets/MBSE2016/hg19/tracks/1M1_H3K27Ac_treat_afterfiting_all.bw</a></p>
</li>
<li>
<p>The hosting server must support byte-range request</p>
</li>
</ul>
</li>
<li>
<p><strong>file</strong></p>
<ul>
<li>
<p>A partial path to a file, such as "path/to/track.gz", relative to the Tumor Portrait root directory</p>
</li>
<li>
<p>ProteinPaint will reject:</p>
<ul>
<li>
<p>absolute path (e.g. /etc/passwd)</p>
</li>
<li>
<p>back-to-previous-level (e.g. ../../etc/passwd)</p>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="track-bedgraph">Track: bedGraph<a class="headerlink" href="#track-bedgraph" title="Permanent link">&para;</a></h3>
<div class="admonition example">
<p class="admonition-title">bedGraph track example for RNA-seq</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;rnaseq&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;config&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;RNA-seq coverage&quot;</span><span class="p">,</span>
        <span class="s2">&quot;tktype&quot;</span><span class="o">:</span><span class="s2">&quot;bedgraph&quot;</span><span class="p">,</span>
        <span class="s2">&quot;pcolor&quot;</span><span class="o">:</span><span class="s2">&quot;#006600&quot;</span><span class="p">,</span>
        <span class="s2">&quot;pcolor2&quot;</span><span class="o">:</span><span class="s2">&quot;#CC00B8&quot;</span><span class="p">,</span>
        <span class="s2">&quot;percentile&quot;</span><span class="o">:</span><span class="mi">95</span>
    <span class="p">},</span>
    <span class="s2">&quot;patient1&quot;</span><span class="o">:</span> <span class="p">{</span>
        <span class="s2">&quot;diagnosis&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;aml/SJAML040582_R1.gz&quot;</span>
        <span class="p">},</span>
        <span class="s2">&quot;relapse&quot;</span><span class="o">:</span><span class="p">{</span>
            <span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;somewhere/xxxxx.gz&quot;</span>
        <span class="p">}</span>
    <span class="p">},</span>
    <span class="s2">&quot;patient2&quot;</span><span class="o">:</span> <span class="p">{</span> <span class="p">...</span> <span class="p">},</span>
    <span class="p">...</span> <span class="nx">more</span> <span class="nx">patients</span> <span class="p">...</span> 
<span class="p">},</span>
</code></pre></div>

</div>
<h3 id="track-bigwig">Track: bigWig<a class="headerlink" href="#track-bigwig" title="Permanent link">&para;</a></h3>
<div class="admonition example">
<p class="admonition-title">bigWig example for RNA-seq</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;rnaseq&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;config&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;RNA-seq coverage&quot;</span><span class="p">,</span>
        <span class="s2">&quot;tktype&quot;</span><span class="o">:</span><span class="s2">&quot;bigwig&quot;</span><span class="p">,</span>
        <span class="s2">&quot;pcolor&quot;</span><span class="o">:</span><span class="s2">&quot;#006600&quot;</span><span class="p">,</span>
        <span class="s2">&quot;pcolor2&quot;</span><span class="o">:</span><span class="s2">&quot;#CC00B8&quot;</span><span class="p">,</span>
        <span class="s2">&quot;percentile&quot;</span><span class="o">:</span><span class="mi">95</span>
    <span class="p">},</span>
    <span class="s2">&quot;patient1&quot;</span><span class="o">:</span> <span class="p">{</span>
        <span class="s2">&quot;sampletype&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;aml/SJAML040582_R1.bw&quot;</span>
        <span class="p">}</span>
    <span class="p">},</span>
    <span class="p">...</span> <span class="nx">more</span> <span class="p">...</span>  
<span class="p">},</span>
</code></pre></div>

</div>
<h3 id="track-numeric2-a-pair-of-numerical-tracks">Track: numeric2 (a pair of numerical tracks)<a class="headerlink" href="#track-numeric2-a-pair-of-numerical-tracks" title="Permanent link">&para;</a></h3>
<p>Overlaying of two numerical tracks, noted by "track1" and "track2".</p>
<ul>
<li>
<p>Track 1</p>
<ul>
<li>Displayed on foreground, axis on left</li>
</ul>
</li>
<li>
<p>Track 2</p>
<ul>
<li>Displayed on background, axis on right</li>
</ul>
</li>
</ul>
<div class="admonition example">
<p class="admonition-title"><strong>rnaseq</strong> coverage-FPKM overlay example</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;rnaseq&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;config&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;RNA-seq&quot;</span><span class="p">,</span>
        <span class="s2">&quot;tktype&quot;</span><span class="o">:</span><span class="s2">&quot;numeric2&quot;</span><span class="p">,</span>
    <span class="s2">&quot;track1&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;coverage&quot;</span><span class="p">,</span>
        <span class="s2">&quot;rangelimit&quot;</span><span class="o">:</span><span class="mi">10000000</span><span class="p">,</span>
        <span class="s2">&quot;pcolor&quot;</span><span class="o">:</span><span class="s2">&quot;#006600&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;pcolor2&quot;</span><span class="o">:</span><span class="s2">&quot;#CC00B8&quot;</span><span class="p">,</span> 
        <span class="s2">&quot;percentile&quot;</span><span class="o">:</span><span class="mi">95</span>
        <span class="p">},</span>
    <span class="s2">&quot;track2&quot;</span><span class="o">:</span><span class="p">{</span> 
        <span class="s2">&quot;pcolor&quot;</span><span class="o">:</span><span class="s2">&quot;#FF9900&quot;</span><span class="p">,</span>
        <span class="s2">&quot;pcolor2&quot;</span><span class="o">:</span><span class="s2">&quot;#CC7A00&quot;</span><span class="p">,</span>
        <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;FPKM&quot;</span><span class="p">,</span>
        <span class="s2">&quot;autoscale&quot;</span><span class="o">:</span><span class="s2">&quot;on&quot;</span>
        <span class="p">}</span>  
    <span class="p">},</span>
    <span class="s2">&quot;patient1&quot;</span><span class="o">:</span> <span class="p">{</span>
        <span class="s2">&quot;diagnosis&quot;</span><span class="o">:</span><span class="p">{</span>
            <span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;aml/SJAML040582_R1.gz&quot;</span><span class="p">,</span>
            <span class="s2">&quot;file2&quot;</span><span class="o">:</span><span class="s2">&quot;aml/10-PAPAIZ-diagnosis.gz&quot;</span>
    <span class="p">}</span>
    <span class="p">},</span> 
    <span class="s2">&quot;patient2&quot;</span><span class="o">:</span> <span class="p">{</span> <span class="p">...</span> <span class="p">},</span>
    <span class="p">...</span> <span class="nx">more</span> <span class="nx">patients</span> <span class="p">...</span> 
<span class="p">},</span>
</code></pre></div>

</div>
<h3 id="track-junction-rna-seq-junction-reads">Track: junction (RNA-seq junction reads)<a class="headerlink" href="#track-junction-rna-seq-junction-reads" title="Permanent link">&para;</a></h3>
<div class="admonition example">
<p class="admonition-title"><strong>junction</strong> example, along with a "browserview" trigger for displaying</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;junction&quot;</span><span class="o">:</span> <span class="p">{</span>
<span class="s2">&quot;config&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;type2color&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;known&quot;</span><span class="o">:</span><span class="s2">&quot;#9C9C9C&quot;</span><span class="p">,</span>
        <span class="s2">&quot;novel&quot;</span><span class="o">:</span><span class="s2">&quot;#cc0000&quot;</span>
    <span class="p">}</span>
<span class="p">},</span>
<span class="s2">&quot;30-PAIFXV&quot;</span><span class="o">:</span> <span class="p">{</span>
    <span class="s2">&quot;diagnosis&quot;</span><span class="o">:</span> <span class="p">{</span>
    <span class="s2">&quot;file&quot;</span><span class="o">:</span> <span class="s2">&quot;junction/targetNBL/30-PAIFXV-diagnosis-SJNBL017066_D1.gz&quot;</span>
    <span class="p">}</span>
<span class="p">},</span>
<span class="s2">&quot;30-PAIPGU&quot;</span><span class="o">:</span> <span class="p">{</span>
    <span class="s2">&quot;diagnosis&quot;</span><span class="o">:</span> <span class="p">{</span>
    <span class="s2">&quot;file&quot;</span><span class="o">:</span> <span class="s2">&quot;junction/targetNBL/30-PAIPGU-diagnosis-SJNBL017070_D1.gz&quot;</span>
    <span class="p">}</span>
<span class="p">},</span>
<span class="p">...</span>
<span class="p">},</span>
<span class="s2">&quot;browserview&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;position&quot;</span><span class="o">:</span><span class="p">{</span> <span class="s2">&quot;chr&quot;</span><span class="o">:</span><span class="s2">&quot;chr12&quot;</span><span class="p">,</span><span class="s2">&quot;start&quot;</span><span class="o">:</span><span class="mi">25357723</span><span class="p">,</span><span class="s2">&quot;stop&quot;</span><span class="o">:</span><span class="mi">25403865</span>    <span class="p">},</span>
    <span class="s2">&quot;assays&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;junction&quot;</span><span class="o">:</span><span class="p">{</span>
            <span class="s2">&quot;sum_view&quot;</span><span class="o">:</span><span class="p">{</span>
                <span class="s2">&quot;type&quot;</span><span class="o">:</span><span class="s2">&quot;junction&quot;</span><span class="p">,</span>
                <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;NBL&quot;</span>
            <span class="p">}</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</code></pre></div>

</div>
<p>Each junction track file contains splicing junctions identified in ONE
SAMPLE only. Primarily this is converted from the RNApeg output
(<a href="http://hc-wiki.stjude.org/display/compbio/How+to+count+novel+or+reference+junction+reads+in+an+RNA-Seq+BAM+using+RNApeg">http://hc-wiki.stjude.org/display/compbio/How+to+count+novel+or+reference+junction+reads+in+an+RNA-Seq+BAM+using+RNApeg</a>).</p>
<p>To convert an RNApeg output file into a track file to be displayed on
ProteinPaint, run:</p>
<div class="highlight"><pre><span></span><code>node utils/rnapegjunction2tabix.js sample.RNApeg.output sample.junction
</code></pre></div>

<p>The converted file has 5 columns:</p>
<ol>
<li>
<p>Chromosome name, e.g. "chr1"</p>
</li>
<li>
<p>Start, 1-based position of the last exon nucleotide</p>
</li>
<li>
<p>Stop, 1-based position of the first exon nucleotide</p>
</li>
<li>
<p>Number of (high-quality) junction reads</p>
</li>
<li>
<p>Type of junction</p>
<p>a.  RNApeg outputs two types of junction (known/novel). For displaying, arbitrary types can be used. Any used types should be stated in the <strong>config.jtype2color</strong> so they can be distinguished by color</p>
</li>
</ol>
<h3 id="track-vcf">Track: VCF<a class="headerlink" href="#track-vcf" title="Permanent link">&para;</a></h3>
<div class="admonition example">
<p class="admonition-title"><strong>VCF</strong> examples, along with "browserview" trigger for displaying</p>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;vcf_cohort&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;vcf/scd_143_join.vcf.gz&quot;</span><span class="p">,</span>
    <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;Joint&quot;</span>
<span class="p">},</span>
<span class="s2">&quot;vcf&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;SJSCD040763&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;G1&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;vcf/SJSCD040763_G1.vcf.gz&quot;</span>
    <span class="p">}</span>
    <span class="p">},</span>
    <span class="s2">&quot;SJSCD040764&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;G1&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;vcf/SJSCD040764_G1.vcf.gz&quot;</span><span class="p">}},</span>
    <span class="p">...</span>
<span class="p">},</span>
<span class="s2">&quot;browserview&quot;</span><span class="o">:</span><span class="p">{</span>
    <span class="s2">&quot;position&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;chr&quot;</span><span class="o">:</span><span class="s2">&quot;chr6&quot;</span><span class="p">,</span><span class="s2">&quot;start&quot;</span><span class="o">:</span><span class="mi">135179289</span><span class="p">,</span><span class="s2">&quot;stop&quot;</span><span class="o">:</span><span class="mi">135183700</span><span class="p">},</span>
    <span class="s2">&quot;assays&quot;</span><span class="o">:</span><span class="p">{</span>
        <span class="s2">&quot;vcf_cohort&quot;</span><span class="o">:</span><span class="mi">1</span><span class="p">,</span>
        <span class="s2">&quot;vcf&quot;</span><span class="o">:</span><span class="p">{</span>
            <span class="s2">&quot;sum_view&quot;</span><span class="o">:</span><span class="p">{</span>
                <span class="s2">&quot;type&quot;</span><span class="o">:</span><span class="s2">&quot;vcf&quot;</span><span class="p">,</span>
                <span class="s2">&quot;name&quot;</span><span class="o">:</span><span class="s2">&quot;Individual&quot;</span>
            <span class="p">}</span>
        <span class="p">}</span>
    <span class="p">}</span>
<span class="p">}</span>
</code></pre></div>

</div>
<p>"Vcf_cohort" specifies one single VCF file.</p>
<p>"Vcf" specifies the VCF file from each sample.</p>
<h3 id="track-vafs1-variant-allele-fraction-of-a-single-sample">Track: vafs1 (variant allele fraction of a single sample)<a class="headerlink" href="#track-vafs1-variant-allele-fraction-of-a-single-sample" title="Permanent link">&para;</a></h3>
<div class="admonition example">
<p class="admonition-title"><strong>vafs1</strong> example</p>
</div>
<div class="highlight"><pre><span></span><code><span class="s2">&quot;vafs1&quot;</span><span class="o">:</span><span class="p">{</span>
  <span class="s2">&quot;person1&quot;</span><span class="o">:</span><span class="p">{</span>
     <span class="s2">&quot;sampletype1&quot;</span><span class="o">:</span><span class="p">{</span>
       <span class="s2">&quot;DNA&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;variantgene/SJALL040467_D1.combine.WGS.goodmarkers.txt.gz&quot;</span><span class="p">},</span>
       <span class="s2">&quot;RNA&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;variantgene/SJALL040467_D1.combine.RNAseq.goodmarkers.txt.gz&quot;</span><span class="p">}</span>
     <span class="p">},</span>
     <span class="s2">&quot;sampletype2&quot;</span><span class="o">:</span><span class="p">{</span>
       <span class="s2">&quot;DNA&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;variantgene/SJALL040467_R1.combine.WGS.goodmarkers.txt.gz&quot;</span><span class="p">},</span>
       <span class="s2">&quot;RNA&quot;</span><span class="o">:</span><span class="p">{</span><span class="s2">&quot;file&quot;</span><span class="o">:</span><span class="s2">&quot;variantgene/SJALL040467_R1.combine.RNAseq.goodmarkers.txt.gz&quot;</span><span class="p">}</span>
     <span class="p">}</span>
  <span class="p">},</span>
  <span class="p">...</span> <span class="nx">next</span> <span class="nx">person</span> <span class="p">...</span> 
<span class="p">}</span>
</code></pre></div>

<p>Track file has 6 columns:</p>
<ol>
<li>
<p>Chromosome</p>
</li>
<li>
<p>Position (1-based)</p>
</li>
<li>
<p>Reference allele</p>
</li>
<li>
<p>alternative allele</p>
</li>
<li>
<p>Total reads</p>
</li>
<li>
<p>Variant allele fraction</p>
</li>
</ol>
<p>Note that for one sample type, there can be multiple types of tracks
(DNA/RNA in above example). The type names can be arbitrary.</p>
<h3 id="track-bam">Track: BAM<a class="headerlink" href="#track-bam" title="Permanent link">&para;</a></h3>
<p>In progress ...</p>
                
                  
                
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../embedding-proteinpaint/" title="Embedding ProteinPaint" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Embedding ProteinPaint
              </span>
            </div>
          </a>
        
        
          <a href="../../../studies/" title="About" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                About
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2020 St. Jude Children's Research Hospital
          </div>
        
        
        built from <a href="https://github.com/stjudecloud/docs/tree/22bb778">22bb778</a> using
        
        <a href="https://www.mkdocs.org">MkDocs</a>
        and a modified version of
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs
        </a>.
      </div>
      
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../../../assets/fonts/font-awesome.css">
    
      <a href="https://www.facebook.com/stjuderesearch/" target="_blank" rel="noopener" title="facebook" class="md-footer-social__link fa fa-facebook"></a>
    
      <a href="https://twitter.com/stjuderesearch?lang=en" target="_blank" rel="noopener" title="twitter" class="md-footer-social__link fa fa-twitter"></a>
    
      <a href="https://www.linkedin.com/company/st-jude-childrens-research-hospital" target="_blank" rel="noopener" title="linkedin" class="md-footer-social__link fa fa-linkedin"></a>
    
      <a href="https://github.com/stjude" target="_blank" rel="noopener" title="github" class="md-footer-social__link fa fa-github"></a>
    
  </div>

    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../../assets/javascripts/application.2d55c750.js"></script>
      
      <script>app.initialize({version:"1.1",url:{base:"../../../.."}})</script>
      
    
  </body>
</html>