## Overview

**Single-End Analysis SEQuencing pipeline** (abbreviated as **SEASEQ**) is a comprehensive automated computational pipeline for all ChIP-Seq/CUT&RUN data analysis.

SEASEQ performs extensive analyses from the raw output of the experiment, including alignment, peak calling, motif analysis, read coverage profiling, clustered peak (e.g. super-enhancer) identification, and quality assessment metrics, as well as automatic interfacing with data in [GEO]/[SRA]. The easy-to-use and versatile format of SEASEQ makes it a reliable and efficient resource for ensuring high quality ChIP-Seq analysis.


[SRA]: https://www.ncbi.nlm.nih.gov/sra
[GEO]: https://www.ncbi.nlm.nih.gov/geo/

## Inputs

|   Name                        |   Type                | Description                                                 |  Example            |
|-------------------------------|-----------------------|-------------------------------------------------------------|---------------------|
|   FASTQ files                 |   Array of files      |   One or more FASTQ files. The files should be gzipped.     |   [`*.gz`]          |
|   SRA identifiers (SRR)       |   Array of strings    |   One of more SRRs.                                         |   [`SRR12345678`]   |
|   Genome Reference            |   File                |   The genome reference.                                     |   [`*.fa`]          |
|   Genome Bowtie indexes       |   Array of files      |   The genome bowtie v1 indexes. Should be six index files.  |   [`*.ebwt`]        |
|   Gene Annotation             |   File                |   The gene annotation.                                      |   [`*.gtf`]         |
|   Blacklists                  |   File                |   UHS/DER/DAC or custom blacklists regions file.            |   [`*/bed`]         |
|   MEME motif databases        |   Array of files      |   One or more of the [MEME suite motif databases]           |   [`*.meme`]        |

### Input configuration

SEASEQ supports FASTQ files and SRA identifiers (SRRs) as input. A combination of both is also supported. 
Bowtie genomic indexes and region-based blacklists are optional.

[MEME suite motif databases]: https://meme-suite.org/meme/db/motifs

## Output

SEASEQ provides multiple outputs from the different analyses offers. 
Outputs are grouped into subdirectories:

| Name                  | Type    | Description                                                                            |
|-----------------------|---------|----------------------------------------------------------------------------------------|
|  BAM_Density          | Folder  | Reads Coverage profiling in promoter and genic regions matrices, plots and heatmaps.   |
|  BAM_files            | Folder  | All mapping (.bam) files.                                                              |
|  MOTIFS               | Folder  | Motifs discovery and prediction results files.                                         |
|  PEAKS                | Folder  | Identified narrow peaks, broad peaks and linear-stitched peaks files.                  |
|  PEAKS_Annotation     | Folder  | Genic annotation of peaks tables and plot.                                             |
|  PEAKS_Display        | Folder  | Normalized signal data tracks in wiggle, tdf and bigwig formats.                       |
|  [QC]                 | Folder  | Quality statistics and metrics of FASTQs and peaks as tables and color-coded HTML.     |

[QC]: #qc-metrics

## Workflow Steps
1. SRRs are downloaded as FASTQs using the [SRA Toolkit](http://www.ncbi.nlm.nih.gov/books/NBK158900/) if provided.

2. Sequencing FASTQs are aligned to the reference genome using [Bowtie](https://doi.org/10.1186/gb-2009-10-3-r25).
3. Mapped reads are further processed by removal of redundant reads and blacklisted regions. 
4. Read density profiling in relevant genomic regions such as promoters and gene body usig [BamToGFF](https://github.com/stjude/BAM2GFF).
5. Normalized and unnormalized coverage files for display are generated for the [UCSC genome browser](https://doi.org/10.1093/bib/bbs038) and [IGV](https://doi.org/10.1093/bib/bbs017). 
6. Identification of enriched regions for two binding profiles:
    * For factors that bind shorter regions, e.g. many sequence-specific transcription factors using [MACS](https://doi.org/10.1186/gb-2008-9-9-r137).
    * For broad regions of enrichment, e.g. some histone modifications using [SICER](https://doi.org/10.1093/bioinformatics/btp340).
7. Identification of stitched clusters of enriched regions and separates exceptionally large regions, e.g. super-enhancers from typical enhancers, using [ROSE](http://younglab.wi.mit.edu/super_enhancer_code.html).
8. Characterization of overrepresentated sequences by motif discovery and enrichment using tools from the [MEME Suite](https://doi.org/10.1093/nar/gkv416).
9. Genic annotation of peaks including promoters, gene bodies, gene-centric windows, and proximal genes.
10. Assessment of quality by calculating relevant metrics including those recommmended by the [ENCODE consortium](https://doi.org/10.1101/gr.136184.111). More information is provided [here](#qc-metrics).

## Creating a workspace

Before you can run one of our workflows, you must first create a workspace in
DNAnexus for the run. Refer to [the general workflow
guide](running-sj-workflows.md#getting-started) to learn how to create a
DNAnexus workspace for each workflow run.

You can navigate to the SEASEQ workflow page
[here](https://platform.stjude.cloud/workflows/seaseq).

## Uploading Input Files

SEASEQ requires at least the genome reference, gene annotation and motif database 
files to be uploaded as [input](#inputs).

Refer to [the general workflow
guide](running-sj-workflows.md#uploading-files) to learn how to upload input
files to the workspace you just created.

## Running the Workflow

Refer to [the general workflow
guide](running-sj-workflows.md#running-the-workflow) to learn how to launch
the workflow, hook up input files, adjust parameters, start a run, and
monitor run progress.

SEASEQ has special preset options in the "Launch Tool" dropdown.
These are explained below:

## Analysis of Results

Each tool in St. Jude Cloud produces a visualization that makes understanding 
results more accessible than working with spreadsheets or tab-delimited files. 
This is a recommended way to view the provided visualization files.

Refer to [the general workflow
guide](running-sj-workflows.md#custom-visualizations) to learn how to access
these visualizations.

We also include the raw output files for you to dig into if the visualization 
is not sufficient to answer your research question.

Refer to [the general workflow
guide](running-sj-workflows.md#raw-results-files) to learn how to access raw
results files.

SEASEQ results will be in the parent `/` folder unless otherwise specified. 

##  Interpreting results

Upon successful run of SEASEQ, all files are saved to the results directory as 
specified in [outputs](#output).

### QC Metrics

SEASEQ provides a vast set of quality metrics for detecting experimental issues, including ChIP-Seq 
metrics recommended by the ENCODE consortium. 

SEASEQ QC metrics include the percentage of reads mapped, uniquely mapped reads (NRF), read 
quality, sequence diversity, fraction of reads in peaks (FRiP), strand correlation scores 
(NSC, RSC), library complexity (PCR bottleneck) and many other important metrics used to 
infer quality. 

In addition, we devised a five-scale color-rank flag system to visually identify excellent 
(score = 2), good (score = 1), average (score = 0), below-average (score = -1) or poor (score = -2) 
results for each metric and a cross-metric summary score (between -2 and 2), using recommended 
thresholds where possible. 
The results are exported in a tab-delimited file and color flagged in an html format.

## Frequently asked questions

If you have any questions not covered here, feel free to reach out
on [our contact
form](https://hospital.stjude.org/apps/forms/fb/st-jude-cloud-contact/).

# References

None yet!


## Similar Topics

[Running our Workflows](../analyzing-data/running-sj-workflows.md)

[Working with our Data Overview](../managing-data/working-with-our-data.md)

[Upload/Download Data (local)](../managing-data/upload-local.md)


[seaseq]: https://github.com/stjude/seaseq
