



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    <script src="//assets.adobedtm.com/launch-EN942f79c83ff84c10a45b4856a85eaccb.min.js" async></script>
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-107924368-1"></script>
    <script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","UA-107924368-1")</script>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
      
        <link rel="canonical" href="https://www.stjude.cloud/docs/guides/tools/rapid-rnaseq/">
      
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../../assets/images/favicon.ico">
      <meta name="generator" content="mkdocs-0.17.4, mkdocs-material-2.9.2">
    
    
      
        <title>Rapid RNA-Seq Fusion Detection - St. Jude Cloud Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application.4eccddb4.css">
      
      
    
    
      <script src="../../../assets/javascripts/modernizr.20ef595d.js"></script>

      <!-- Adobe Analytics -->
      <script src="//assets.adobedtm.com/launch-EN942f79c83ff84c10a45b4856a85eaccb.min.js" async></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
    
    <link rel="stylesheet" href="../../../assets/fonts/material-icons.css">
    
      <link rel="stylesheet" href="../../../css/extra.css">
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="__github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="__drawer"></label>
    
      <a href="#overview" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://www.stjude.cloud/docs/" title="St. Jude Cloud Documentation" class="md-header-nav__button md-logo">
          
            <img src="../../../images/stjude-logo-child-white.png" width="40" height="28">
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                St. Jude Cloud Documentation
              </span>
              <span class="md-header-nav__topic">
                Rapid RNA-Seq Fusion Detection
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="__search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="__search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/stjude/sjcloud-docs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      stjude/sjcloud-docs
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="__drawer">
    <span class="md-nav__button md-logo">
      
        <img src="../../../images/stjude-logo-child-white.png" width="48" height="48">
      
    </span>
    St. Jude Cloud Documentation
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/stjude/sjcloud-docs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#__github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      stjude/sjcloud-docs
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Welcome to St. Jude Cloud" class="md-nav__link">
      Welcome to St. Jude Cloud
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../create-an-account/" title="Create An Account" class="md-nav__link">
      Create An Account
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Data Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Data Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/types-of-data/" title="Types of Data" class="md-nav__link">
      Types of Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/metadata/" title="Metadata Provided" class="md-nav__link">
      Metadata Provided
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/data-request/" title="Making a Data Request" class="md-nav__link">
      Making a Data Request
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/working-with-our-data/" title="Working with Our Data" class="md-nav__link">
      Working with Our Data
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/creating-a-cloud-app/" title="Creating a Cloud App" class="md-nav__link">
      Creating a Cloud App
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/data-transfer-app/" title="Data Transfer Application" class="md-nav__link">
      Data Transfer Application
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data/command-line/" title="Command Line Interaction" class="md-nav__link">
      Command Line Interaction
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked>
    
    <label class="md-nav__link" for="nav-4">
      Tool Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Tool Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../pecan-pie/" title="Pecan PIE" class="md-nav__link">
      Pecan PIE
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../neoepitope/" title="NeoepitopePred" class="md-nav__link">
      NeoepitopePred
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../chipseq/" title="ChIP-Seq Peak Calling" class="md-nav__link">
      ChIP-Seq Peak Calling
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="__toc">
        Rapid RNA-Seq Fusion Detection
      </label>
    
    <a href="./" title="Rapid RNA-Seq Fusion Detection" class="md-nav__link md-nav__link--active">
      Rapid RNA-Seq Fusion Detection
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-started" title="Getting started" class="md-nav__link">
    Getting started
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-data" title="Uploading data" class="md-nav__link">
    Uploading data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-tool" title="Running the tool" class="md-nav__link">
    Running the tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hooking-up-inputs" title="Hooking up inputs" class="md-nav__link">
    Hooking up inputs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starting-the-workflow" title="Starting the workflow" class="md-nav__link">
    Starting the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monitoring-run-progress" title="Monitoring run progress" class="md-nav__link">
    Monitoring run progress
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-of-results" title="Analysis of results" class="md-nav__link">
    Analysis of results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#finding-the-raw-results-files" title="Finding the raw results files" class="md-nav__link">
    Finding the raw results files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#navigating-results" title="Navigating Results" class="md-nav__link">
    Navigating Results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#raw-result-files" title="Raw result files" class="md-nav__link">
    Raw result files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#custom-visualization-results" title="Custom visualization results" class="md-nav__link">
    Custom visualization results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-results" title="Interpreting results" class="md-nav__link">
    Interpreting results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#known-issues" title="Known issues" class="md-nav__link">
    Known issues
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" title="Frequently asked questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../warden/" title="WARDEN Differential Expression Analysis" class="md-nav__link">
      WARDEN Differential Expression Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mutational-signatures/" title="Mutational Signatures" class="md-nav__link">
      Mutational Signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cis-x/" title="" class="md-nav__link">
      
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Visualization Guides
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Visualization Guides
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/pecan/" title="PeCan Data Portal" class="md-nav__link">
      PeCan Data Portal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/genome-paint/" title="Genome Paint" class="md-nav__link">
      Genome Paint
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../portals/sickle-cell/" title="Sickle Cell Genomics Portal" class="md-nav__link">
      Sickle Cell Genomics Portal
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Forms
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Forms
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../forms/how-to-fill-out-DAA/" title="Filling out a Data Access Agreement" class="md-nav__link">
      Filling out a Data Access Agreement
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../forms/how-to-fill-out-Extension/" title="Filling out an Extension Addendum" class="md-nav__link">
      Filling out an Extension Addendum
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../citation-reference/" title="Citing St. Jude Cloud" class="md-nav__link">
      Citing St. Jude Cloud
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../faq/" title="Frequently Asked Questions" class="md-nav__link">
      Frequently Asked Questions
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../glossary/" title="Glossary" class="md-nav__link">
      Glossary
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="__toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" title="Overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-started" title="Getting started" class="md-nav__link">
    Getting started
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#uploading-data" title="Uploading data" class="md-nav__link">
    Uploading data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-tool" title="Running the tool" class="md-nav__link">
    Running the tool
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hooking-up-inputs" title="Hooking up inputs" class="md-nav__link">
    Hooking up inputs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starting-the-workflow" title="Starting the workflow" class="md-nav__link">
    Starting the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#monitoring-run-progress" title="Monitoring run progress" class="md-nav__link">
    Monitoring run progress
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysis-of-results" title="Analysis of results" class="md-nav__link">
    Analysis of results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#finding-the-raw-results-files" title="Finding the raw results files" class="md-nav__link">
    Finding the raw results files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#navigating-results" title="Navigating Results" class="md-nav__link">
    Navigating Results
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#raw-result-files" title="Raw result files" class="md-nav__link">
    Raw result files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#custom-visualization-results" title="Custom visualization results" class="md-nav__link">
    Custom visualization results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#interpreting-results" title="Interpreting results" class="md-nav__link">
    Interpreting results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#known-issues" title="Known issues" class="md-nav__link">
    Known issues
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#frequently-asked-questions" title="Frequently asked questions" class="md-nav__link">
    Frequently asked questions
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/stjude/sjcloud-docs/edit/master/docs/guides/tools/rapid-rnaseq.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                  <h1>Rapid RNA-Seq Fusion Detection</h1>
                
                <table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Authors</strong></td>
<td>Scott Newman, Clay McLeod, Yongjin Li</td>
</tr>
<tr>
<td><strong>Publication</strong></td>
<td>N/A (not published)</td>
</tr>
<tr>
<td><strong>Technical Support</strong></td>
<td><a href="https://stjude.cloud/contact">Contact Us</a></td>
</tr>
</tbody>
</table>
<p>Fusion genes are important for cancer diagnosis, subtype definition and
targeted therapy. RNASeq is useful for detecting fusion transcripts; however, computational methods face challenges to identify fusion transcripts
due to events such as internal tandem duplication (ITD), multiple genes, low
expression, or non-templated insertions. To address some of these challenges, St. Jude Cloud offers "Rapid RNA-Seq", an end-to-end
clinically validated pipeline that detects gene fusions and ITDs from human RNA-Seq.</p>
<h2 id="overview">Overview</h2>
<p><strong>Inputs</strong></p>
<p>The input can be either of the two entries below, based on whether you want to start
with FastQ files or a BAM file.</p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>Paired FastQ files</td>
<td>Gzipped FastQ files generated by human RNA-Seq</td>
<td>Sample_R1.fastq.gz and Sample_R2.fastq.gz</td>
</tr>
<tr>
<td>BAM file</td>
<td>Aligned reads file from human RNA-Seq</td>
<td>Sample.bam</td>
</tr>
</tbody>
</table>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>If you provide a BAM file to the pipeline, it <strong>must</strong> be aligned to GRCh37-lite.
Running a BAM aligned to any other reference genome is not supported. Maybe more
importantly, we do not check the genome build of the BAM, so errors in computation
or the results can occur. If your BAM is <em>not</em> aligned to this genome build, we 
recommend converting the BAM back to FastQ files using 
<a href="https://broadinstitute.github.io/picard/command-line-overview.html#SamToFastq">Picard's SamToFastq</a>
functionality and using the FastQ version of the pipeline.</p>
</div>
<p><strong>Outputs</strong></p>
<p>The Rapid RNA-Seq pipeline produces the following outputs:</p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>Predicted gene fusions (.txt)</td>
<td>File containing putative gene fusions.</td>
</tr>
<tr>
<td>Coverage file (.bw)</td>
<td>bigWig file containing coverage information.</td>
</tr>
<tr>
<td>Splice junction read counts (.txt)</td>
<td>Read counts for the splice junction detected.</td>
</tr>
<tr>
<td>Interactive fusion visualization</td>
<td>Fusion visualization produced by ProteinPaint.</td>
</tr>
<tr>
<td>Interactive coverage visualization</td>
<td>Coverage visualization produced by ProteinPaint.</td>
</tr>
</tbody>
</table>
<p><strong>Process</strong></p>
<ol>
<li>The raw sequence data is aligned to GRCh37-lite using standard STAR
   mapping.</li>
<li>A coverage bigWig (.bw) file is produced to allow the user to assess
   sample quality across the genome.</li>
<li>Two gene fusion detection algorithms are run in parallel.<ul>
<li>The <strong>Fuzzion</strong> (Rice et al. unpublished data) fusion detection
  algorithm is run to provide high sensitivity for recurrent gene
  fusions.</li>
<li>The <strong>RNAPEG</strong> (Edmonson et al. unpublished data) splice
  junction read counting algorithm is run to quantify read counts
  for splice junctions. These splice junction read counts are then
  used by <strong>Cicero</strong> (Li et al. unpublished data) to detect
  putative gene fusions.</li>
</ul>
</li>
<li>Custom visualizations for putative gene fusions and genome coverage
   are produced by ProteinPaint.</li>
</ol>
<p><strong>Mapping</strong></p>
<p>We use the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905/">STAR aligner</a> 
to rapidly map reads to the GRCh37 human reference genome. This step generally 
takes around one hour to complete assuming approximately 55-75 million paired reads
are supplied.</p>
<p><strong>Coverage</strong></p>
<p>Internally developed scripts calculate the coverage of mapped reads
genome wide. The resulting bigWig file can be viewed in ProteinPaint or
used for quality control.</p>
<p><strong>Splice junction read quantification</strong></p>
<p>We use our RNAPEG software to quantify reads spanning known and novel
splice junctions. RNAPEG also corrects improper mappings at splice
junction boundaries for more accurate definition of novel splice
junctions. The resulting junctions file can be viewed along with the
coverage bigWig file to gain insights into gene expression and splicing
patterns</p>
<p><strong>Genome-wide fusion prediction</strong></p>
<p>We developed an assembly-based algorithm CICERO (Clipped-reads Extended
for RNA Optimization) that is able to extend the read-length spanning
fusion junctions for detecting complex fusions. CICERO finds clipped
reads and junction spanning reads, assembles them into a contig and maps
the contig back to the reference genome. Mapped contigs are then
annotated and filtered. Those with potential genic effects including
gene fusion, ITD, readthrough or circular RNA are reported in the
<code class="codehilite"><span class="n">final_fusions</span><span class="p">.</span><span class="n">txt</span></code> file. An interactive version of this file with
predictions sorted by quality can be inspected with the ProteinPaint
interactive fusion viewer.</p>
<p>An abstract describing CICERO was presented at ASHG, 2014:
<a href="http://www.ashg.org/2014meeting/abstracts/fulltext/f140120024.htm">http://www.ashg.org/2014meeting/abstracts/fulltext/f140120024.htm</a></p>
<p><strong>Low stringency fusion gene "Hotpot" search</strong></p>
<p>We have observed that certain fusions such as KIAA1549-BRAF in low-grade
glioma have apparently limited read support in the bam file â€” either due
to low expression or low tumor purity. In these cases, we use a
secondary tool, FUZZION, that performs fuzzy matching for known fusion
gene junctions for every read in the bam file (both mapped and
unmapped). FUZZION can recover even a single low quality read
potentially supporting a known fusion gene junction. The FUZZION output
is a simple text file with read IDs and sequences supporting a
particular gene fusion. The fusion point is indicated with square
brackets <code class="codehilite"><span class="p">[]</span></code>.</p>
<h2 id="getting-started">Getting started</h2>
<div class="admonition caution">
<p class="admonition-title">Caution</p>
<p>This pipeline assumes GRCh37-lite coordinates. If your BAM is 
<em>not</em> aligned to this genome build, we recommend converting the BAM 
back to FastQ files using <a href="https://broadinstitute.github.io/picard/command-line-overview.html#SamToFastq">Picard's SamToFastq</a>
functionality.</p>
</div>
<p>To get started, you need to navigate to the <a href="https://platform.stjude.cloud/tools/rapid_rna-seq">Rapid RNA-Seq tool page</a>. You'll need to click
the "Start" button in the left hand pane. This creates a cloud workspace
in DNAnexus with the same name as the tool. After this, you will be able 
to upload your input files to that workspace.</p>
<p><img alt="" src="../../../images/guides/tools/rapid-rnaseq/click-start.gif" /></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If you can't see the "Start" button, one of these two scenarios is likely the case:</p>
<ul>
<li>You see three buttons on the left sidebar instead of one. In this case,
  you've already clicked the "Start" button previously, and a cloud workspace has
  already been created for you. In this case, you're good! You can move
  on to the next section.</li>
<li>If you cannot see <em>any</em> buttons on the left side, you probably have not
  logged in yet. If you see a sentence that says "Log in to launch this 
  tool", simply login and try again.</li>
</ul>
<p>If neither of these are the case and you still can't click "Start",
<a href="https://stjude.cloud/contact">contact us</a>.</p>
</div>
<h2 id="uploading-data">Uploading data</h2>
<p>The Rapid RNA-Seq pipeline takes either a paired set of Gzipped FastQ files or 
a GRCh37-lite aligned BAM from human RNA-Seq. You can upload your input file(s)
using the <a href="../../data/data-transfer-app/">data transfer application</a>
or by uploading them through <a href="../../data/command-line/">the command line</a>.
Both of the guides linked here will contain more details on how to upload
data using that method, so we defer to those guides here.</p>
<h2 id="running-the-tool">Running the tool</h2>
<p>Once you've uploaded data to your cloud workspace, click "Launch Tool" on the <a href="https://platform.stjude.cloud/tools/rapid_rna-seq">tool's landing page</a>. A dropdown will present the different presets for running the workflow. Here, you can select whether you wish to start with FastQ files or a BAM file.</p>
<p><img alt="" src="../../../images/guides/tools/rapid-rnaseq/launch-tool.gif" /></p>
<h3 id="hooking-up-inputs">Hooking up inputs</h3>
<p>Next, you'll need to hook up the FastQ files you uploaded in 
<a href="#uploading-data">the upload data section</a>. In this example,
we are using the FastQ version of the pipeline, so you can 
hook up the inputs by clicking on the <code class="codehilite"><span class="n">Fastq</span><span class="o">/</span><span class="n">R1</span></code> and <code class="codehilite"><span class="n">Fastq</span><span class="o">/</span><span class="n">R2</span></code>
slots and selecting the respective files. If you are using 
the BAM-based workflow, the process is similar.</p>
<p><img alt="" src="../../../images/guides/tools/rapid-rnaseq/hookup-inputs.gif" /></p>
<h3 id="starting-the-workflow">Starting the workflow</h3>
<p>Once your input files are hooked up, you should be able to start the workflow
by clicking the "Run as Analysis..." button in the top right hand corner
of the workflow dialog.</p>
<p><img alt="" src="../../../images/guides/tools/rapid-rnaseq/run-analysis.gif" /></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>If you cannot click this button, please ensure that all of the inputs are correctly hooked up (see <a href="#hooking-up-inputs">hooking up inputs</a>).</p>
<p>If you're still have trouble, please <a href="https://stjude.cloud/contact">contact us</a> and include
a screenshot of the workflow screen above.</p>
</div>
<h2 id="monitoring-run-progress">Monitoring run progress</h2>
<p>Once you have started one or more Rapid RNA-Seq runs, you can safely close your
browser and come back later to check the status of the jobs. To do this,
navigate to the <a href="https://platform.stjude.cloud/tools/rapid_rna-seq">tool's landing page</a>. 
Next, click "View Results" then select the "View Running Jobs" option. 
You will be redirected to the job monitoring page. Each job you kicked off
gets one row in this table.</p>
<p><img alt="" src="../../../images/guides/tools/rapid-rnaseq/monitoring-jobs.gif" /> </p>
<p>You can click the "+" on any of the runs to check 
the status of individual steps of the Rapid RNA-Seq pipeline.
Other information, such as time, cost of individual steps in the pipeline, 
and even viewing the job logs can accessed by clicking around the sub-items.</p>
<p><img alt="" src="../../../images/guides/tools/rapid-rnaseq/job-detailed-view.gif" /> </p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Power users can view the <a href="https://wiki.dnanexus.com/UI/Jobs">DNAnexus Job Monitoring Tutorial</a> and the <a href="https://wiki.dnanexus.com/Command-Line-Client/Monitoring-Executions">DNAnexus Command Line Tutorial for Job Monitoring</a> for advanced capabilities for monitoring jobs.</p>
</div>
<h2 id="analysis-of-results">Analysis of results</h2>
<p>Each tool in St. Jude Cloud produces a visualization that makes understanding
results more accessible than working with Excel spreadsheets or tab delimited
files. This is the primary way we recommend you work with your Rapid RNA-Seq results. We also
include the raw output files for you to dig into if the visualization is not 
sufficient to answer your research question.</p>
<h3 id="finding-the-raw-results-files">Finding the raw results files</h3>
<p>Navigate to the <a href="https://platform.stjude.cloud/tools/rapid_rna-seq">tool's landing page</a>. 
In the left hand pane, click "View Results" then "View Results Files". You will
be taken to the filesystem view your cloud workspace. This is similar to your the
filesystem on your computer, and you can do many common operations such as deleting,
renaming, and moving files.</p>
<p><img alt="" src="../../../images/guides/tools/rapid-rnaseq/rapid-rna-results.png" /> </p>
<h3 id="navigating-results">Navigating Results</h3>
<h4 id="raw-result-files">Raw result files</h4>
<p>Navigate to your tool's description page (for instance, Rapid RNA-Seq's
description page is
<a href="https://platform.stjude.cloud/tools/rapid_rna-seq">here</a>). You should
see a screen similar to the one in the screenshot below. In the left
hand pane, select "View Results Files".</p>
<p><img alt="Click &quot;View Results Files&quot;" src="../../../images/guides/tools/common/raw-results.png" /></p>
<p>You should now be in the tool's workspace with access to files that you
uploaded and results files that are generated. How/where the result
files are generated are specific to each pipeline. Please refer to your
individual pipeline's documentation on where the output files are kept.</p>
<p><img alt="" src="../../../images/guides/tools/common/raw-results-success.png" /></p>
<h4 id="custom-visualization-results">Custom visualization results</h4>
<p>Navigate to your tool's description page (for instance, Rapid RNA-Seq's
description page is
<a href="https://platform.stjude.cloud/tools/rapid_rna-seq">here</a>). You should
see a screen similar to the one in the screenshot below. In the left
hand pane, select "Visualize Results".</p>
<p><img alt="Click &quot;Visualize Results&quot;" src="../../../images/guides/tools/common/visualize-results.png" /></p>
<p>You should now see a list of visualization files like the ones shown below.</p>
<p><img alt="" src="../../../images/guides/tools/common/visualize-results-success.png" /></p>
<h3 id="interpreting-results">Interpreting results</h3>
<p>The complete output file specification is listed in the <a href="#overview">overview section</a>
of this guide. Here, we will discuss each of the different output files in more detail.</p>
<ul>
<li><strong>Predicted gene fusions</strong>: The putative gene fusions will be
  contained in the file <code class="codehilite"><span class="o">[</span><span class="n">SAMPLE</span><span class="o">]</span><span class="p">.</span><span class="n">final_fusions</span><span class="p">.</span><span class="n">txt</span><span class="w"></span></code>. This file is a
  tab-delimited file containing many fields for each of the predicted
  SV. The most important columns are the following.</li>
</ul>
<table>
<thead>
<tr>
<th>Field Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code class="codehilite"><span class="n">sample</span></code></td>
<td>Sample name</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">gene</span><span class="o">*</span></code></td>
<td>Gene name</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">chr</span><span class="o">*</span></code></td>
<td>Chromosome name</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">pos</span><span class="o">*</span></code></td>
<td>Genomic Location</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">ort</span><span class="o">*</span></code></td>
<td>Strand</td>
</tr>
<tr>
<td><code class="codehilite"><span class="k">reads</span><span class="o">*</span></code></td>
<td>Supporting reads</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">medal</span></code></td>
<td>Estimated pathogenicity assessment using St. Jude Medal Ceremony</td>
</tr>
</tbody>
</table>
<ul>
<li><strong>Coverage file</strong>: A standard
    <a href="https://genome.ucsc.edu/goldenpath/help/bigWig.html">bigWig</a> file
    used to describe genomic read coverage.</li>
<li><strong>Splice junction read counts</strong>: A custom file format describing the
    junction read counts. The following fields are included in the
    tab-delimited output file.</li>
</ul>
<table>
<thead>
<tr>
<th>Field Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code class="codehilite"><span class="n">junction</span></code></td>
<td>Splice junction in the TCGA format "chrX:start:+,chrX:end,+". "start" and "end" are the 1-based position of the last mapped nucleotide before the skip and the first mapped nucleotide after the skip (i.e. the last base of the previous exon and the first base of the next exon). Note that in <code class="codehilite"><span class="na">.bed</span></code> output these coordinates will be different, see the .bed output section below. The "+" is currently hardcoded, though this may change in the future.</td>
</tr>
<tr>
<td><code class="codehilite"><span class="k">count</span></code></td>
<td>Raw count of reads supporting the junction. During correction counts for ambiguous junctions can be combined, though obviously these additional reads will not be visible in the raw BAM file.</td>
</tr>
<tr>
<td><code class="codehilite"><span class="k">type</span></code></td>
<td>Either "known" (matching a reference junction) or "novel" (not observed in the reference junction collection).</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">genes</span></code></td>
<td>Gene symbols from the junction calling process. These still need work in the raw junction calling process, it's recommended to use the "annotated" output files instead which assign matching HUGO gene symbols based on the UCSC refGene table.</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">transcripts</span></code></td>
<td>List of known transcript IDs matching the junction.</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">qc_flanking</span></code></td>
<td>Count of supporting reads passing flanking sequence checks (junctions observed adjacent to read ends require 18+ nt of flanking sequence, otherwise 10+ nt).</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">qc_plus</span></code></td>
<td>Count of supporting reads aligned to the + strand.</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">qc_minus</span></code></td>
<td>Count of supporting reads aligned to the - strand.</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">qc_perfect_reads</span></code></td>
<td>Count of supporting reads with perfect alignments (no reference mismatches of quality 15+, indels, or soft clips).</td>
</tr>
<tr>
<td><code class="codehilite"><span class="n">qc_clean_reads</span></code></td>
<td>Count of supporting reads whose alignments are not perfect but which have a ratio of &lt;= 5% of reference mismatches of quality 15+, indels, or soft clips relative to the count of aligned bases on both the left and right flanking sequence. Note: qc_clean_reads does NOT include qc_perfect_reads: to get a count of "perfect plus pretty good" reads the two values must be added together.</td>
</tr>
</tbody>
</table>
<h2 id="known-issues">Known issues</h2>
<div class="admonition caution">
<p class="admonition-title">Adapter contamination</p>
<p>This pipeline does not, at present, remove adapter sequences. If the
sequencing library is contaminated with adapters, CICERO runtimes can
increase exponentially. We recommend running FastQ files through a QC
pipeline such as FastQC and trimming adapters with tools such as
Trimmomatic if adapters are found.</p>
</div>
<div class="admonition caution">
<p class="admonition-title">High coverage regions</p>
<p>Certain cell types show very high transcription of certain loci, for
example, the immunoglobulin heavy chain locus in plasma cells. The
presence of very highly covered regions (typically 100,000-1,000,000+ X)
has an adverse effect on CICERO runtimes. Presently, we have no good
solution to this problem as strategies such as down-sampling may reduce
sensitivity over important regions of the genome.</p>
</div>
<div class="admonition bug">
<p class="admonition-title">Interactive Visualizations Exon vs Intron Nomenclature</p>
<p>When a codon is split over a fusion gene junction, the annotation
software marks the event as intronic when really, the event should be
exonic. We are working to fix this bug. In the mean time, if a fusion is
predicted to be in frame but the interactive plot shows "intronic", we
suggest the user blat the contig shown just below to clarify if the true
junction is either in the intron or exon.</p>
</div>
<h2 id="frequently-asked-questions">Frequently asked questions</h2>
<p>None yet! If you have any questions not covered here, feel free to reach
out on <a href="https://hospital.stjude.org/apps/forms/fb/st-jude-cloud-contact/">our contact form</a>.</p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../chipseq/" title="ChIP-Seq Peak Calling" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                ChIP-Seq Peak Calling
              </span>
            </div>
          </a>
        
        
          <a href="../warden/" title="WARDEN Differential Expression Analysis" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                WARDEN Differential Expression Analysis
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2017 - 2019 St. Jude Children's Research Hospital
          </div>
        
        
        built from <a href="https://github.com/stjude/sjcloud-docs/tree/c16b33d">c16b33d</a> using
        
        <a href="https://www.mkdocs.org">MkDocs</a>
        and a modified version of
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>.
      </div>
      
        
  <div class="md-footer-social">
    <link rel="stylesheet" href="../../../assets/fonts/font-awesome.css">
    
      <a href="https://www.facebook.com/stjuderesearch/" class="md-footer-social__link fa fa-facebook"></a>
    
      <a href="https://twitter.com/stjuderesearch?lang=en" class="md-footer-social__link fa fa-twitter"></a>
    
      <a href="https://www.linkedin.com/company/st-jude-childrens-research-hospital" class="md-footer-social__link fa fa-linkedin"></a>
    
      <a href="https://github.com/stjude" class="md-footer-social__link fa fa-github"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application.ec5be3af.js"></script>
      
      <script>app.initialize({version:"0.17.4",url:{base:"../../.."}})</script>
      
    
    
      
    
  </body>
</html>